// @vue-macros/define-prop@0.1.6/dist/chunk-UNIDB5IY.mjs
// src/core/utils.ts
function stringifyArray(strs) {
  return `[${strs.map((s) => JSON.stringify(s)).join(", ")}]`;
}

// src/core/kevin-edition.ts
import { DEFINE_PROP } from "@vue-macros/common";
import { genRuntimePropDefinition } from "@vue-macros/api";
var kevinEdition = ({ s, offset, resolveTSType }) => {
  const props = [];
  return {
    walkCall(node, parent) {
      var _a;
      const [name, definition] = node.arguments;
      let propName;
      if (!name) {
        if (parent.type !== "VariableDeclarator" || parent.id.type !== "Identifier")
          throw new Error(
            `A variable must be used to receive the return value of ${DEFINE_PROP} when the first argument is not passed. (kevinEdition)`
          );
        propName = parent.id.name;
      } else if (name.type !== "StringLiteral") {
        throw new Error(
          `The first argument of ${DEFINE_PROP} must be a literal string. (kevinEdition)`
        );
      } else {
        propName = name.value;
      }
      props.push({
        name: propName,
        definition: definition ? s.sliceNode(definition, { offset }) : void 0,
        typeParameter: (_a = node.typeParameters) == null ? void 0 : _a.params[0]
      });
      return propName;
    },
    async genRuntimeProps(isProduction) {
      if (props.length === 0)
        return;
      const isAllWithoutOptions = props.every(
        ({ definition, typeParameter }) => !definition && !typeParameter
      );
      if (isAllWithoutOptions) {
        return stringifyArray(props.map(({ name }) => name));
      }
      let propsString = "{\n";
      for (const { name, definition, typeParameter } of props) {
        let def;
        const types = typeParameter && await resolveTSType(typeParameter);
        if (definition && !types) {
          def = definition;
        } else {
          const properties = [];
          if (definition)
            properties.push(`...${definition}`);
          def = genRuntimePropDefinition(types, isProduction, properties);
        }
        propsString += `  ${JSON.stringify(name)}: ${def},
`;
      }
      propsString += "}";
      return propsString;
    }
  };
};

// src/core/johnson-edition.ts
import { DEFINE_PROP as DEFINE_PROP2 } from "@vue-macros/common";
import { genRuntimePropDefinition as genRuntimePropDefinition2 } from "@vue-macros/api";
var johnsonEdition = ({ s, offset, resolveTSType }) => {
  const props = [];
  return {
    walkCall(node, parent) {
      var _a;
      const [value, required, rest] = node.arguments;
      if (parent.type !== "VariableDeclarator" || parent.id.type !== "Identifier")
        throw new Error(
          `A variable must be used to receive the return value of ${DEFINE_PROP2} (johnsonEdition)`
        );
      const propName = parent.id.name;
      props.push({
        name: propName,
        value: value ? s.sliceNode(value, { offset }) : void 0,
        required: required ? s.sliceNode(required, { offset }) : void 0,
        rest: rest ? s.sliceNode(rest, { offset }) : void 0,
        typeParameter: (_a = node.typeParameters) == null ? void 0 : _a.params[0]
      });
      return propName;
    },
    async genRuntimeProps(isProduction) {
      if (props.length === 0)
        return;
      const isAllWithoutOptions = props.every(
        ({ typeParameter, value, required, rest }) => !typeParameter && !value && !required && !rest
      );
      if (isAllWithoutOptions) {
        return stringifyArray(props.map(({ name }) => name));
      }
      let propsString = "{\n";
      for (const { name, value, required, rest, typeParameter } of props) {
        let def;
        const types = typeParameter && await resolveTSType(typeParameter);
        if (rest && !value && !required && !types) {
          def = rest;
        } else {
          const properties = [];
          if (value)
            properties.push(`default: ${value}`);
          if (required)
            properties.push(`required: ${required}`);
          if (rest)
            properties.push(`...${rest}`);
          def = genRuntimePropDefinition2(types, isProduction, properties);
        }
        propsString += `  ${JSON.stringify(name)}: ${def},
`;
      }
      propsString += "}";
      return propsString;
    }
  };
};

// src/core/index.ts
import {
  DEFINE_PROP as DEFINE_PROP3,
  DEFINE_PROPS,
  HELPER_PREFIX,
  MagicString,
  getTransformResult,
  importHelperFn,
  isCallOf,
  parseSFC,
  walkAST
} from "@vue-macros/common";
import { inferRuntimeType, resolveTSReferencedType } from "@vue-macros/api";
var PROPS_VARIABLE_NAME = `${HELPER_PREFIX}props`;
async function transformDefineProp(code, id, edition = "kevinEdition", isProduction = false) {
  if (!code.includes(DEFINE_PROP3))
    return;
  const { scriptSetup, getSetupAst } = parseSFC(code, id);
  if (!scriptSetup)
    return;
  const setupAst = getSetupAst();
  const offset = scriptSetup.loc.start.offset;
  const s = new MagicString(code);
  const { walkCall, genRuntimeProps } = (edition === "kevinEdition" ? kevinEdition : johnsonEdition)({ s, offset, resolveTSType });
  let hasDefineProps = false;
  let hasDefineProp = false;
  walkAST(setupAst, {
    enter(node, parent) {
      if (isCallOf(node, DEFINE_PROP3)) {
        hasDefineProp = true;
        const propName = walkCall(node, parent);
        s.overwriteNode(
          node,
          `${importHelperFn(
            s,
            offset,
            "toRef"
          )}(${PROPS_VARIABLE_NAME}, ${JSON.stringify(propName)})`,
          { offset }
        );
      } else if (isCallOf(node, DEFINE_PROPS)) {
        hasDefineProps = true;
      }
    }
  });
  if (hasDefineProps && hasDefineProp)
    throw new Error(
      `${DEFINE_PROP3} can not be used in the same file as ${DEFINE_PROPS}.`
    );
  const runtimeProps = await genRuntimeProps(isProduction);
  if (runtimeProps)
    s.prependLeft(
      offset,
      `
const ${PROPS_VARIABLE_NAME} = defineProps(${runtimeProps});
`
    );
  return getTransformResult(s, id);
  async function resolveTSType(type) {
    const resolved = await resolveTSReferencedType({
      scope: {
        kind: "file",
        filePath: id,
        content: scriptSetup.content,
        ast: setupAst.body
      },
      type
    });
    return resolved && inferRuntimeType(resolved);
  }
}

export {
  stringifyArray,
  kevinEdition,
  johnsonEdition,
  PROPS_VARIABLE_NAME,
  transformDefineProp
};
