// @vue-macros/define-emit@0.1.5/dist/chunk-EKFZRW6J.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _chunkQFN7HVZGjs = require('./chunk-QFN7HVZG.js');

// src/index.ts
var _unplugin = require('unplugin');






var _common = require('@vue-macros/common');
var _api = require('@vue-macros/api');
function resolveOption(options) {
  const version = options.version || _common.detectVueVersion.call(void 0, );
  return {
    include: [_common.REGEX_VUE_SFC, _common.REGEX_SETUP_SFC, _common.REGEX_VUE_SUB],
    isProduction: process.env.NODE_ENV === "production",
    ...options,
    version
  };
}
var name = "unplugin-vue-define-emit";
var src_default = _unplugin.createUnplugin.call(void 0, 
  (userOptions = {}, { framework }) => {
    const options = resolveOption(userOptions);
    const filter = _common.createFilter.call(void 0, options);
    const { resolve, handleHotUpdate } = _api.RollupResolve.call(void 0, );
    return {
      name,
      enforce: "pre",
      buildStart() {
        if (framework === "rollup" || framework === "vite") {
          _api.setResolveTSFileIdImpl.call(void 0, resolve(this));
        }
      },
      transformInclude(id) {
        return filter(id);
      },
      transform(code, id) {
        return _chunkQFN7HVZGjs.transformDefineEmit.call(void 0, code, id);
      },
      vite: {
        configResolved(config) {
          options.isProduction = config.isProduction;
        },
        handleHotUpdate
      }
    };
  }
);



exports.src_default = src_default;
