// @vue-macros/define-models@1.0.5/dist/chunk-5TWKXEEO.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});






var _chunkS22O3WGFjs = require('./chunk-S22O3WGF.js');

// src/index.ts
var _unplugin = require('unplugin');







var _common = require('@vue-macros/common');
function resolveOption(options) {
  const version = options.version || _common.detectVueVersion.call(void 0, );
  return {
    include: [_common.REGEX_VUE_SFC, _common.REGEX_SETUP_SFC, _common.REGEX_VUE_SUB],
    unified: true,
    ...options,
    version
  };
}
var name = "unplugin-vue-define-models";
var src_default = _unplugin.createUnplugin.call(void 0, 
  (userOptions = {}) => {
    const options = resolveOption(userOptions);
    const filter = _common.createFilter.call(void 0, options);
    return {
      name,
      enforce: "pre",
      resolveId(id) {
        if (id.startsWith(_chunkS22O3WGFjs.helperPrefix))
          return id;
      },
      loadInclude(id) {
        return id.startsWith(_chunkS22O3WGFjs.helperPrefix);
      },
      load(_id) {
        const id = _common.normalizePath.call(void 0, _id);
        if (id === _chunkS22O3WGFjs.emitHelperId)
          return _chunkS22O3WGFjs.emit_helper_default;
        else if (id === _chunkS22O3WGFjs.useVmodelHelperId)
          return _chunkS22O3WGFjs.use_vmodel_default;
      },
      transformInclude(id) {
        return filter(id);
      },
      transform(code, id) {
        return _chunkS22O3WGFjs.transformDefineModels.call(void 0, code, id, options.version, options.unified);
      }
    };
  }
);



exports.src_default = src_default;
