// unplugin-vue-macros@2.2.1/dist/chunk-EV7NWTSX.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; } function _nullishCoalesce(lhs, rhsFn) { if (lhs != null) { return lhs; } else { return rhsFn(); } }// src/index.ts


var _unplugincombine = require('unplugin-combine');
var _common = require('@vue-macros/common');
var _devtools = require('@vue-macros/devtools');
var _betterdefine = require('@vue-macros/better-define'); var _betterdefine2 = _interopRequireDefault(_betterdefine);
var _defineemit = require('@vue-macros/define-emit'); var _defineemit2 = _interopRequireDefault(_defineemit);
var _definemodels = require('@vue-macros/define-models'); var _definemodels2 = _interopRequireDefault(_definemodels);
var _unpluginvuedefineoptions = require('unplugin-vue-define-options'); var _unpluginvuedefineoptions2 = _interopRequireDefault(_unpluginvuedefineoptions);
var _defineprop = require('@vue-macros/define-prop'); var _defineprop2 = _interopRequireDefault(_defineprop);
var _defineprops = require('@vue-macros/define-props'); var _defineprops2 = _interopRequireDefault(_defineprops);
var _definepropsrefs = require('@vue-macros/define-props-refs'); var _definepropsrefs2 = _interopRequireDefault(_definepropsrefs);
var _definerender = require('@vue-macros/define-render'); var _definerender2 = _interopRequireDefault(_definerender);
var _defineslots = require('@vue-macros/define-slots'); var _defineslots2 = _interopRequireDefault(_defineslots);
var _exportexpose = require('@vue-macros/export-expose'); var _exportexpose2 = _interopRequireDefault(_exportexpose);
var _exportprops = require('@vue-macros/export-props'); var _exportprops2 = _interopRequireDefault(_exportprops);
var _hoiststatic = require('@vue-macros/hoist-static'); var _hoiststatic2 = _interopRequireDefault(_hoiststatic);
var _namedtemplate = require('@vue-macros/named-template'); var _namedtemplate2 = _interopRequireDefault(_namedtemplate);
var _reactivitytransform = require('@vue-macros/reactivity-transform'); var _reactivitytransform2 = _interopRequireDefault(_reactivitytransform);
var _setupblock = require('@vue-macros/setup-block'); var _setupblock2 = _interopRequireDefault(_setupblock);
var _setupcomponent = require('@vue-macros/setup-component'); var _setupcomponent2 = _interopRequireDefault(_setupcomponent);
var _setupsfc = require('@vue-macros/setup-sfc'); var _setupsfc2 = _interopRequireDefault(_setupsfc);
var _shortemits = require('@vue-macros/short-emits'); var _shortemits2 = _interopRequireDefault(_shortemits);
function resolveOptions({
  root,
  version,
  plugins,
  isProduction,
  nuxtContext,
  betterDefine,
  defineEmit,
  defineModels,
  defineOptions,
  defineProp,
  defineProps,
  definePropsRefs,
  defineRender,
  defineSlots,
  exportExpose,
  exportProps,
  hoistStatic,
  namedTemplate,
  reactivityTransform,
  setupBlock,
  setupComponent,
  setupSFC,
  shortEmits
}) {
  function resolveSubOptions(options, commonOptions, defaultEnabled = true) {
    options = _nullishCoalesce(options, () => ( defaultEnabled));
    if (!options)
      return false;
    return { ...options === true ? {} : options, ...commonOptions };
  }
  root = root || process.cwd();
  version = version || _common.detectVueVersion.call(void 0, root);
  isProduction = _nullishCoalesce(isProduction, () => ( process.env.NODE_ENV === "production"));
  return {
    plugins: plugins || {},
    root,
    version,
    isProduction,
    nuxtContext: nuxtContext || {},
    betterDefine: resolveSubOptions(betterDefine, {
      version,
      isProduction
    }),
    defineEmit: resolveSubOptions(defineEmit, {
      isProduction,
      version
    }),
    defineModels: resolveSubOptions(defineModels, { version }),
    defineOptions: resolveSubOptions(
      defineOptions,
      { version },
      version < 3.3
    ),
    defineProp: resolveSubOptions(defineProp, {
      isProduction,
      version
    }),
    defineProps: resolveSubOptions(defineProps, { version }),
    definePropsRefs: resolveSubOptions(definePropsRefs, {
      version
    }),
    defineRender: resolveSubOptions(defineRender, { version }),
    defineSlots: resolveSubOptions(
      defineSlots,
      { version },
      version < 3.3
    ),
    exportExpose: resolveSubOptions(
      exportExpose,
      { version },
      false
    ),
    exportProps: resolveSubOptions(
      exportProps,
      { version },
      false
    ),
    hoistStatic: resolveSubOptions(hoistStatic, { version }),
    namedTemplate: resolveSubOptions(namedTemplate, {
      version
    }),
    reactivityTransform: resolveSubOptions(
      reactivityTransform,
      { version }
    ),
    setupBlock: resolveSubOptions(setupBlock, { version }, false),
    setupComponent: resolveSubOptions(setupComponent, {
      version,
      root
    }),
    setupSFC: resolveSubOptions(setupSFC, { version }),
    shortEmits: resolveSubOptions(
      shortEmits,
      { version },
      version < 3.3
    )
  };
}
function resolvePlugin(unplugin, framework, options) {
  if (!options)
    return;
  return unplugin[framework](options);
}
var name = "unplugin-vue-macros";
var src_default = _unplugincombine.createCombinePlugin.call(void 0, 
  (userOptions = {}, meta) => {
    const options = resolveOptions(userOptions);
    const framework = meta.framework;
    const setupComponentPlugins = resolvePlugin(
      _setupcomponent2.default,
      framework,
      options.setupComponent
    );
    const namedTemplatePlugins = resolvePlugin(
      _namedtemplate2.default,
      framework,
      options.namedTemplate
    );
    const plugins = [
      resolvePlugin(_setupsfc2.default, framework, options.setupSFC),
      setupComponentPlugins == null ? void 0 : setupComponentPlugins[0],
      resolvePlugin(_setupblock2.default, framework, options.setupBlock),
      namedTemplatePlugins == null ? void 0 : namedTemplatePlugins[0],
      resolvePlugin(_defineprop2.default, framework, options.defineProp),
      resolvePlugin(_defineemit2.default, framework, options.defineEmit),
      resolvePlugin(_defineprops2.default, framework, options.defineProps),
      resolvePlugin(_definepropsrefs2.default, framework, options.definePropsRefs),
      resolvePlugin(_exportprops2.default, framework, options.exportProps),
      resolvePlugin(_shortemits2.default, framework, options.shortEmits),
      resolvePlugin(_definemodels2.default, framework, options.defineModels),
      resolvePlugin(_defineslots2.default, framework, options.defineSlots),
      resolvePlugin(_exportexpose2.default, framework, options.exportExpose),
      resolvePlugin(
        _reactivitytransform2.default,
        framework,
        options.reactivityTransform
      ),
      resolvePlugin(_betterdefine2.default, framework, options.betterDefine),
      resolvePlugin(_hoiststatic2.default, framework, options.hoistStatic),
      resolvePlugin(_unpluginvuedefineoptions2.default, framework, options.defineOptions),
      options.plugins.vue,
      options.plugins.vueJsx,
      resolvePlugin(_definerender2.default, framework, options.defineRender),
      setupComponentPlugins == null ? void 0 : setupComponentPlugins[1],
      namedTemplatePlugins == null ? void 0 : namedTemplatePlugins[1],
      framework === "vite" ? _devtools.Devtools.call(void 0, { nuxtContext: options.nuxtContext }) : void 0
    ].filter(Boolean);
    return {
      name,
      plugins
    };
  }
);




exports.resolveOptions = resolveOptions; exports.src_default = src_default;
