// @unocss-ui/components@0.0.2-alpha.6/dist/components.es.js
import { defineComponent as C, computed as I, openBlock as u, createElementBlock as d, mergeProps as P, createElementVNode as c, normalizeClass as p, unref as w, createCommentVNode as k, renderSlot as $, toDisplayString as A, createBlock as E, resolveDynamicComponent as ue, withCtx as B, withDirectives as O, isRef as Y, vModelCheckbox as se, ref as U, reactive as F, onMounted as Me, normalizeStyle as ve, watch as q, onUnmounted as tn, Teleport as pt, getCurrentScope as nn, onScopeDispose as an, toRef as ln, readonly as on, customRef as sn, nextTick as bt, createVNode as K, Transition as pe, Fragment as S, renderList as M, withModifiers as D, createTextVNode as _, withKeys as rn, vModelText as un, pushScopeId as $t, popScopeId as wt, vModelDynamic as cn, normalizeProps as Ve, createSlots as dn, vModelRadio as fn, createStaticVNode as xt, TransitionGroup as hn, vShow as kt, useAttrs as gn, resolveComponent as yn, provide as mn, useSlots as vn, guardReactiveProps as pn } from "vue";
const bn = { class: "flex" }, $n = { key: 0 }, wn = {
  key: 0,
  class: "ml-2 h-6 w-6 flex items-center justify-center"
}, xn = {
  name: "UAlert"
}, Tt = /* @__PURE__ */ C({
  ...xn,
  props: {
    type: { default: "primary" },
    icon: { default: "" },
    closeable: { type: Boolean, default: !1 },
    noIcon: { type: Boolean, default: !1 },
    colorful: { type: Boolean, default: !1 },
    bordered: { type: Boolean, default: !1 }
  },
  emits: ["close"],
  setup(e) {
    const t = e, n = I(() => {
      if (t.noIcon)
        return "";
      if (t.icon)
        return t.icon;
      switch (t.type) {
        case "success":
          return "i-mdi:check-circle text-success-500";
        case "info":
          return "i-mdi:information text-info-500";
        case "warning":
          return "i-mdi:alert-circle text-warning-500";
        case "error":
          return "i-mdi:close-circle text-error-500";
        case "primary":
          return "i-mdi:information-slab-circle text-primary-500";
        case "secondary":
          return "i-mdi:information-slab-circle text-secondary-500";
        case "accent":
          return "i-mdi:information-slab-circle text-accent-500";
        default:
          return "";
      }
    });
    return (a, s) => (u(), d("div", P(a.$attrs, {
      class: ["max-w-lg flex justify-between rounded-lg px-4 py-3", [
        e.colorful ? `bg-${e.type}-50 text-${e.type}-800` : "bg-white text-gray-700",
        e.bordered ? "border border-current" : "border-none shadow"
      ]]
    }), [
      c("div", bn, [
        e.noIcon ? k("", !0) : (u(), d("div", $n, [
          c("div", {
            class: p(["mr-3 h-6 w-6", w(n)])
          }, null, 2)
        ])),
        $(a.$slots, "default")
      ]),
      e.closeable ? (u(), d("div", wn, [
        c("button", {
          class: "i-mdi:close-thick h-4 w-4 cursor-pointer transition hover:scale-125",
          onClick: s[0] || (s[0] = (l) => a.$emit("close"))
        })
      ])) : k("", !0)
    ], 16));
  }
}), kn = { class: "relative cursor-default" }, Tn = {
  name: "UBadge"
}, Cn = /* @__PURE__ */ C({
  ...Tn,
  props: {
    type: { default: "default" },
    size: { default: "md" },
    content: { default: "" }
  },
  setup(e) {
    return (t, n) => (u(), d("span", kn, [
      $(t.$slots, "default", {}, void 0, !0),
      c("span", {
        class: p(["absolute translate-x-1/2 rounded-full text-white -right-1 -top-1/2", [
          e.content ? "min-w-5" : "h-3 w-3",
          [e.type],
          {
            "text-xs": e.size === "sm",
            "px-1 py-0.5 text-xs": e.size === "md",
            "px-2.5 py-1 text-sm ": e.size === "lg"
          }
        ]])
      }, A(e.content), 3)
    ]));
  }
});
const ce = (e, t) => {
  const n = e.__vccOpts || e;
  for (const [a, s] of t)
    n[a] = s;
  return n;
}, Vn = /* @__PURE__ */ ce(Cn, [["__scopeId", "data-v-5426b2a5"]]), jn = { class: "flex items-center" }, Sn = {
  name: "UButton",
  inheritAttrs: !1
}, fe = /* @__PURE__ */ C({
  ...Sn,
  props: {
    as: { default: "button" },
    type: { default: "default" },
    size: { default: "md" },
    disabled: { type: Boolean, default: !1 }
  },
  setup(e) {
    return (t, n) => (u(), d("div", jn, [
      (u(), E(ue(e.as), P(t.$attrs, {
        type: "button",
        disabled: e.disabled,
        "aria-disabled": e.disabled,
        class: ["inline-flex cursor-pointer items-center justify-center whitespace-nowrap rounded-md font-sans text-xs font-semibold leading-4 shadow-sm disabled:cursor-not-allowed disabled:opacity-30 focus:outline-none focus:ring-2 focus:ring-offset-2", [
          e.type === "default" ? "text-gray-700 bg-gray-100 hover:bg-gray-200 focus:ring-gray-500" : `text-white bg-${e.type}-500 hover:bg-${e.type}-600 focus:ring-${e.type}-500`,
          {
            "px-2 py-1": e.size === "sm",
            "px-3 py-2": e.size === "md",
            "px-4 py-3": e.size === "lg"
          }
        ]]
      }), {
        default: B(() => [
          $(t.$slots, "default")
        ]),
        _: 3
      }, 16, ["disabled", "aria-disabled", "class"]))
    ]));
  }
});
fe.install = function(e) {
  e.component(fe.name, fe);
};
var An = typeof global == "object" && global && global.Object === Object && global;
const Ct = An;
var In = typeof self == "object" && self && self.Object === Object && self, Bn = Ct || In || Function("return this")();
const L = Bn;
var Pn = L.Symbol;
const R = Pn;
function On(e, t) {
  for (var n = -1, a = e == null ? 0 : e.length, s = Array(a); ++n < a; )
    s[n] = t(e[n], n, e);
  return s;
}
var Mn = Array.isArray;
const be = Mn;
var Vt = Object.prototype, Un = Vt.hasOwnProperty, zn = Vt.toString, ae = R ? R.toStringTag : void 0;
function En(e) {
  var t = Un.call(e, ae), n = e[ae];
  try {
    e[ae] = void 0;
    var a = !0;
  } catch {
  }
  var s = zn.call(e);
  return a && (t ? e[ae] = n : delete e[ae]), s;
}
var Ln = Object.prototype, Dn = Ln.toString;
function _n(e) {
  return Dn.call(e);
}
var Kn = "[object Null]", Fn = "[object Undefined]", Ge = R ? R.toStringTag : void 0;
function Q(e) {
  return e == null ? e === void 0 ? Fn : Kn : Ge && Ge in Object(e) ? En(e) : _n(e);
}
function ee(e) {
  return e != null && typeof e == "object";
}
var Nn = "[object Symbol]";
function Rn(e) {
  return typeof e == "symbol" || ee(e) && Q(e) == Nn;
}
var Gn = 1 / 0, He = R ? R.prototype : void 0, qe = He ? He.toString : void 0;
function jt(e) {
  if (typeof e == "string")
    return e;
  if (be(e))
    return On(e, jt) + "";
  if (Rn(e))
    return qe ? qe.call(e) : "";
  var t = e + "";
  return t == "0" && 1 / e == -Gn ? "-0" : t;
}
function Hn(e) {
  return e == null ? "" : jt(e);
}
var qn = 0;
function X(e) {
  var t = ++qn;
  return Hn(e) + t;
}
const Wn = { class: "h-6 flex items-center" }, Zn = ["for"], Yn = ["id", "disabled", "value"], Xn = ["for"], Jn = {
  name: "UCheckbox"
}, Qn = /* @__PURE__ */ C({
  ...Jn,
  props: {
    modelValue: { default: !1 },
    type: { default: "primary" },
    size: { default: "md" },
    checked: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    value: { default: null },
    checkedValue: { default: !0 },
    uncheckedValue: { default: !1 },
    labelLeft: { type: Boolean, default: !1 }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const n = e, a = X("checkbox-"), s = I({
      get: () => n.value ? n.modelValue || n.value : n.checkedValue && n.modelValue === n.checkedValue ? !0 : n.uncheckedValue && n.modelValue === n.uncheckedValue ? !1 : n.modelValue || n.checked,
      set: (l) => {
        if (n.value) {
          t("update:modelValue", l);
          return;
        }
        if (n.checkedValue && l) {
          t("update:modelValue", n.checkedValue);
          return;
        } else if (n.uncheckedValue && !l) {
          t("update:modelValue", n.uncheckedValue);
          return;
        }
        t("update:modelValue", l);
      }
    });
    return (l, o) => (u(), d("div", Wn, [
      l.$slots.default && e.labelLeft ? (u(), d("label", {
        key: 0,
        for: w(a),
        class: p(["mr-2 select-none", e.disabled ? "cursor-not-allowed" : "cursor-pointer"])
      }, [
        $(l.$slots, "default")
      ], 10, Zn)) : k("", !0),
      O(c("input", P({
        id: w(a),
        "onUpdate:modelValue": o[0] || (o[0] = (r) => Y(s) ? s.value = r : null)
      }, l.$attrs, {
        type: "checkbox",
        disabled: e.disabled,
        value: e.value,
        class: ["border-gray-300 rounded", [
          `text-${e.type}-400 focus:ring-${e.type}-500`,
          e.disabled ? "cursor-not-allowed opacity-30" : "cursor-pointer",
          {
            "h-3 w-3": e.size === "sm",
            "h-4 w-4": e.size === "md",
            "h-5 w-5": e.size === "lg"
          }
        ]]
      }), null, 16, Yn), [
        [se, w(s)]
      ]),
      l.$slots.default && !e.labelLeft ? (u(), d("label", {
        key: 1,
        for: w(a),
        class: p(["ml-2 select-none", e.disabled ? "cursor-not-allowed" : "cursor-pointer"])
      }, [
        $(l.$slots, "default")
      ], 10, Xn)) : k("", !0)
    ]));
  }
}), ea = { class: "border border-gray-100 rounded-md p-4" }, ta = { class: "flex-1" }, na = /* @__PURE__ */ c("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  fill: "none",
  viewBox: "0 0 24 24",
  "stroke-width": "1.5",
  stroke: "currentColor",
  class: "w-5 text-gray-500"
}, [
  /* @__PURE__ */ c("path", {
    "stroke-linecap": "round",
    "stroke-linejoin": "round",
    d: "M19.5 8.25l-7.5 7.5-7.5-7.5"
  })
], -1), aa = [
  na
], la = {
  name: "UCollapse"
}, oa = /* @__PURE__ */ C({
  ...la,
  props: {
    expanded: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    iconLeft: { type: Boolean, default: !1 }
  },
  setup(e) {
    const t = e, n = U(null), a = F({
      collapsed: !t.expanded,
      expandedCompleted: !1,
      maxHeight: 0,
      toggleExpand() {
        t.disabled || (a.expandedCompleted = !1, a.collapsed && setTimeout(() => {
          a.expandedCompleted = !0;
        }, 300), a.collapsed = !a.collapsed, a.resizeContent());
      },
      resizeContent() {
        var s;
        a.maxHeight = a.collapsed ? 0 : ((s = n.value) == null ? void 0 : s.offsetHeight) || 0;
      }
    });
    return Me(() => a.resizeContent()), (s, l) => (u(), d("div", ea, [
      c("div", {
        class: p(["flex items-center", [
          e.disabled ? "cursor-not-allowed" : "cursor-pointer",
          {
            "flex-row-reverse": e.iconLeft
          }
        ]]),
        onClick: l[0] || (l[0] = //@ts-ignore
        (...o) => a.toggleExpand && a.toggleExpand(...o))
      }, [
        c("div", ta, [
          $(s.$slots, "default")
        ]),
        c("div", {
          class: p(["transition transition-transform duration-200 ease-in-out", [
            e.iconLeft ? "mr-2" : "ml-2",
            {
              "rotate-180": !a.collapsed
            }
          ]])
        }, aa, 2)
      ], 2),
      c("div", {
        class: p(["transition-all duration-300 ease-in-out", {
          "overflow-y-hidden": !a.expandedCompleted
        }]),
        style: ve({ maxHeight: `${a.maxHeight}px` })
      }, [
        c("div", {
          ref_key: "content",
          ref: n
        }, [
          $(s.$slots, "content")
        ], 512)
      ], 6)
    ]));
  }
}), sa = { class: "relative" }, ra = { class: "absolute inset-0 flex items-center" }, ia = {
  name: "UDivider"
}, ua = /* @__PURE__ */ C({
  ...ia,
  props: {
    align: { default: "center" },
    dashed: { type: Boolean, default: !1 },
    thickness: { default: 2 }
  },
  setup(e) {
    return (t, n) => (u(), d("div", P({ class: "w-full" }, t.$attrs), [
      c("div", sa, [
        c("div", ra, [
          c("div", {
            class: p(["w-full border-t border-t-1 border-gray-300", [`border-t-${e.thickness}`, { "border-dashed": e.dashed }]])
          }, null, 2)
        ]),
        c("div", {
          class: p(["relative flex", {
            "justify-center": e.align === "center",
            "justify-end": e.align === "right",
            "justify-start": e.align === "left"
          }])
        }, [
          $(t.$slots, "default")
        ], 2)
      ])
    ], 16));
  }
}), ca = /* @__PURE__ */ c("div", { class: "absolute inset-0 bg-gray-500 opacity-75" }, null, -1), da = [
  ca
], fa = {
  name: "UDrawer"
}, ha = /* @__PURE__ */ C({
  ...fa,
  props: {
    modelValue: { type: Boolean, default: !1 },
    placement: { default: "left" },
    width: { default: "md" },
    duration: { default: "200" }
  },
  emits: ["update:modelValue", "close"],
  setup(e, { emit: t }) {
    const n = e;
    q(
      () => n.modelValue,
      (s) => {
        s && t("update:modelValue", s), document.body.style.overflow = s ? "hidden" : "auto";
      }
    );
    function a() {
      t("update:modelValue", !1), t("close");
    }
    return tn(() => {
      document.body.style.overflow = "auto";
    }), (s, l) => (u(), E(pt, { to: "body" }, [
      e.modelValue ? (u(), d("div", {
        key: 0,
        class: "fixed inset-0 z-40",
        onClick: a
      }, da)) : k("", !0),
      c("div", P(s.$attrs, {
        class: ["fixed z-50 overflow-y-auto bg-white transition-all duration-200", [`duration-${e.duration}`, {
          [`h-screen w-${e.width}`]: e.placement === "right" || e.placement === "left",
          [`w-full h-${e.width}`]: e.placement === "top" || e.placement === "bottom",
          "-left-full top-0": e.placement === "left" && !e.modelValue,
          "left-0 top-0": e.placement === "left" && e.modelValue,
          "-right-full top-0": e.placement === "right" && !e.modelValue,
          "right-0 top-0": e.placement === "right" && e.modelValue,
          "-top-full left-0": e.placement === "top" && !e.modelValue,
          "top-0 left-0 ": e.placement === "top" && e.modelValue,
          "-bottom-full left-0": e.placement === "bottom" && !e.modelValue,
          "bottom-0 left-0": e.placement === "bottom" && e.modelValue
        }]]
      }), [
        $(s.$slots, "default")
      ], 16)
    ]));
  }
});
function St(e) {
  return nn() ? (an(e), !0) : !1;
}
function ye(e) {
  return typeof e == "function" ? e() : w(e);
}
const At = typeof window < "u", Ue = () => {
}, je = /* @__PURE__ */ ga();
function ga() {
  var e;
  return At && ((e = window == null ? void 0 : window.navigator) == null ? void 0 : e.userAgent) && /* @__PURE__ */ /iP(ad|hone|od)/.test(window.navigator.userAgent);
}
const We = {
  mounted: "mounted",
  updated: "updated",
  unmounted: "unmounted"
};
function ya(...e) {
  if (e.length !== 1)
    return ln(...e);
  const t = e[0];
  return typeof t == "function" ? on(sn(() => ({ get: t, set: Ue }))) : U(t);
}
function le(e) {
  var t;
  const n = ye(e);
  return (t = n == null ? void 0 : n.$el) != null ? t : n;
}
const It = At ? window : void 0;
function he(...e) {
  let t, n, a, s;
  if (typeof e[0] == "string" || Array.isArray(e[0]) ? ([n, a, s] = e, t = It) : [t, n, a, s] = e, !t)
    return Ue;
  Array.isArray(n) || (n = [n]), Array.isArray(a) || (a = [a]);
  const l = [], o = () => {
    l.forEach((y) => y()), l.length = 0;
  }, r = (y, x, m, h) => (y.addEventListener(x, m, h), () => y.removeEventListener(x, m, h)), i = q(
    () => [le(t), ye(s)],
    ([y, x]) => {
      o(), y && l.push(
        ...n.flatMap((m) => a.map((h) => r(y, m, h, x)))
      );
    },
    { immediate: !0, flush: "post" }
  ), f = () => {
    i(), o();
  };
  return St(f), f;
}
let Ze = !1;
function Ye(e, t, n = {}) {
  const { window: a = It, ignore: s = [], capture: l = !0, detectIframe: o = !1 } = n;
  if (!a)
    return;
  je && !Ze && (Ze = !0, Array.from(a.document.body.children).forEach((m) => m.addEventListener("click", Ue)));
  let r = !0;
  const i = (m) => s.some((h) => {
    if (typeof h == "string")
      return Array.from(a.document.querySelectorAll(h)).some((g) => g === m.target || m.composedPath().includes(g));
    {
      const g = le(h);
      return g && (m.target === g || m.composedPath().includes(g));
    }
  }), y = [
    he(a, "click", (m) => {
      const h = le(e);
      if (!(!h || h === m.target || m.composedPath().includes(h))) {
        if (m.detail === 0 && (r = !i(m)), !r) {
          r = !0;
          return;
        }
        t(m);
      }
    }, { passive: !0, capture: l }),
    he(a, "pointerdown", (m) => {
      const h = le(e);
      h && (r = !m.composedPath().includes(h) && !i(m));
    }, { passive: !0 }),
    o && he(a, "blur", (m) => {
      var h;
      const g = le(e);
      ((h = a.document.activeElement) == null ? void 0 : h.tagName) === "IFRAME" && !(g != null && g.contains(a.document.activeElement)) && t(m);
    })
  ].filter(Boolean);
  return () => y.forEach((m) => m());
}
const ze = {
  [We.mounted](e, t) {
    const n = !t.modifiers.bubble;
    if (typeof t.value == "function")
      e.__onClickOutside_stop = Ye(e, t.value, { capture: n });
    else {
      const [a, s] = t.value;
      e.__onClickOutside_stop = Ye(e, a, Object.assign({ capture: n }, s));
    }
  },
  [We.unmounted](e) {
    e.__onClickOutside_stop();
  }
};
function Bt(e) {
  const t = window.getComputedStyle(e);
  if (t.overflowX === "scroll" || t.overflowY === "scroll" || t.overflowX === "auto" && e.clientHeight < e.scrollHeight || t.overflowY === "auto" && e.clientWidth < e.scrollWidth)
    return !0;
  {
    const n = e.parentNode;
    return !n || n.tagName === "BODY" ? !1 : Bt(n);
  }
}
function ma(e) {
  const t = e || window.event, n = t.target;
  return Bt(n) ? !1 : t.touches.length > 1 ? !0 : (t.preventDefault && t.preventDefault(), !1);
}
function va(e, t = !1) {
  const n = U(t);
  let a = null, s;
  q(ya(e), (r) => {
    if (r) {
      const i = r;
      s = i.style.overflow, n.value && (i.style.overflow = "hidden");
    }
  }, {
    immediate: !0
  });
  const l = () => {
    const r = ye(e);
    !r || n.value || (je && (a = he(
      r,
      "touchmove",
      (i) => {
        ma(i);
      },
      { passive: !1 }
    )), r.style.overflow = "hidden", n.value = !0);
  }, o = () => {
    const r = ye(e);
    !r || !n.value || (je && (a == null || a()), r.style.overflow = s, n.value = !1);
  };
  return St(o), I({
    get() {
      return n.value;
    },
    set(r) {
      r ? l() : o();
    }
  });
}
function pa() {
  let e = !1;
  const t = U(!1);
  return (n, a) => {
    if (t.value = a.value, e)
      return;
    e = !0;
    const s = va(n, a.value);
    q(t, (l) => s.value = l);
  };
}
pa();
const ba = { class: "py-1" }, $a = { class: "px-1 py-2 text-sm" }, wa = ["onClick"], xa = { class: "ml-2" }, ka = {
  key: 1,
  class: "my-2 border dark:border-slate-600"
}, Ta = {
  name: "UDropdown"
}, Ca = /* @__PURE__ */ C({
  ...Ta,
  props: {
    options: { default: () => [] },
    trigger: { default: "click" }
  },
  emits: ["select"],
  setup(e, { emit: t }) {
    const n = e, a = U(null), s = U(null), l = F({
      show: !1,
      timeout: void 0,
      showPanel() {
        l.show = !0, clearTimeout(l.timeout), bt(() => {
          if (a.value && s.value) {
            const o = a.value.getBoundingClientRect(), r = window.innerHeight / 2, i = window.innerWidth / 2;
            o.top > r ? s.value.style.bottom = "calc(100% + 0.5rem)" : s.value.classList.add("mt-2"), o.right > i ? s.value.classList.add("right-0") : s.value.classList.add("left-0");
          }
        });
      },
      closePanel() {
        l.timeout = setTimeout(
          () => l.show = !1,
          n.trigger === "hover" ? 200 : 0
        );
      },
      select(o) {
        o.disabled || (l.show = !1, l.timeout = void 0, t("select", o));
      }
    });
    return (o, r) => O((u(), d("div", {
      class: "relative inline-block text-left",
      onMouseleave: r[3] || (r[3] = () => e.trigger === "hover" && l.closePanel())
    }, [
      c("div", {
        ref_key: "targetRef",
        ref: a,
        class: "w-full inline-flex items-center justify-center rounded-md",
        onMouseenter: r[0] || (r[0] = () => e.trigger === "hover" && l.showPanel()),
        onClick: r[1] || (r[1] = () => e.trigger === "click" && l.showPanel())
      }, [
        $(o.$slots, "default")
      ], 544),
      K(pe, {
        "enter-active-class": "transition duration-200 ease-out",
        "enter-from-class": "translate-y-1 opacity-0",
        "enter-to-class": "translate-y-0 opacity-100",
        "leave-active-class": "transition duration-150 ease-in",
        "leave-from-class": "translate-y-0 opacity-100",
        "leave-to-class": "translate-y-1 opacity-0"
      }, {
        default: B(() => [
          l.show ? (u(), d("div", {
            key: 0,
            ref_key: "dropdownRef",
            ref: s,
            class: "absolute z-10 min-w-max rounded-lg bg-white shadow-lg dark:bg-slate-800",
            tabindex: "-1",
            onMouseenter: r[2] || (r[2] = //@ts-ignore
            (...i) => l.showPanel && l.showPanel(...i))
          }, [
            c("div", ba, [
              c("div", $a, [
                (u(!0), d(S, null, M(e.options, (i) => (u(), d(S, {
                  key: i.value
                }, [
                  i ? (u(), d("div", {
                    key: 0,
                    class: p(["flex cursor-pointer items-center rounded-md px-3 py-1", [
                      i.disabled ? "text-gray-700 opacity-50 cursor-not-allowed" : "hover:bg-primary-200 hover:text-primary-600"
                    ]]),
                    onClick: D((f) => l.select(i), ["stop"])
                  }, [
                    typeof i.icon == "string" ? (u(), d("div", {
                      key: 0,
                      class: p([i.icon, "w-5 text-gray-500"])
                    }, null, 2)) : (u(), E(ue(i.icon), {
                      key: 1,
                      class: "w-5"
                    })),
                    c("span", xa, A(i.label), 1)
                  ], 10, wa)) : (u(), d("div", ka))
                ], 64))), 128))
              ])
            ])
          ], 544)) : k("", !0)
        ]),
        _: 1
      })
    ], 32)), [
      [w(ze), () => e.trigger === "click" && l.closePanel()]
    ]);
  }
}), Va = { class: "flex items-center" }, ja = { class: "flex items-center justify-center" }, Sa = {
  key: 0,
  class: "flex"
}, Aa = /* @__PURE__ */ c("line", {
  x1: "18",
  y1: "6",
  x2: "6",
  y2: "18"
}, null, -1), Ia = /* @__PURE__ */ c("line", {
  x1: "6",
  y1: "6",
  x2: "18",
  y2: "18"
}, null, -1), Ba = [
  Aa,
  Ia
], Pa = {
  name: "UTag"
}, Se = /* @__PURE__ */ C({
  ...Pa,
  props: {
    type: { default: "default" },
    size: { default: "md" },
    rounded: { default: "sm" },
    border: { type: Boolean, default: !1 },
    closeable: { type: Boolean, default: !1 }
  },
  emits: ["close"],
  setup(e) {
    return (t, n) => (u(), d("div", Va, [
      c("div", {
        class: p(["block cursor-default whitespace-nowrap text-xs font-medium shadow-sm", [
          `rounded-${e.rounded}`,
          e.type === "default" ? "bg-gray-100 text-gray-800" : `bg-${e.type}-100 text-${e.type}-800`,
          {
            "border border-gray-300": e.border,
            "px-2 py-1": e.size === "sm",
            "px-3 py-1.5": e.size === "md",
            "px-6 py-3": e.size === "lg"
          }
        ]])
      }, [
        c("div", ja, [
          e.closeable ? (u(), d("div", Sa, [
            $(t.$slots, "default"),
            (u(), d("svg", {
              class: "ml-1 h-4 w-4 cursor-pointer stroke-2 transition-colors duration-150 ease-in-out hover:text-gray-700",
              width: "24",
              height: "24",
              viewBox: "0 0 24 24",
              stroke: "currentColor",
              "stroke-linejoin": "round",
              "stroke-linecap": "round",
              fill: "none",
              role: "presentation",
              onClick: n[0] || (n[0] = (a) => t.$emit("close"))
            }, Ba))
          ])) : $(t.$slots, "default", { key: 1 })
        ])
      ], 2)
    ]));
  }
}), Oa = { class: "flex flex-wrap items-center" }, Ma = { class: "relative h-4 min-w-7" }, Ua = ["id"], za = ["for", "textContent"], Ea = /* @__PURE__ */ c("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  width: "21",
  height: "21",
  viewBox: "0 0 24 24"
}, [
  /* @__PURE__ */ c("path", {
    fill: "#888888",
    d: "M11 17h2v-4h4v-2h-4V7h-2v4H7v2h4v4Zm1 5q-2.075 0-3.9-.788t-3.175-2.137q-1.35-1.35-2.137-3.175T2 12q0-2.075.788-3.9t2.137-3.175q1.35-1.35 3.175-2.137T12 2q2.075 0 3.9.788t3.175 2.137q1.35 1.35 2.138 3.175T22 12q0 2.075-.788 3.9t-2.137 3.175q-1.35 1.35-3.175 2.138T12 22Zm0-2q3.35 0 5.675-2.325T20 12q0-3.35-2.325-5.675T12 4Q8.65 4 6.325 6.325T4 12q0 3.35 2.325 5.675T12 20Zm0-8Z"
  })
], -1), La = {
  name: "UDynamicTags"
}, Da = /* @__PURE__ */ C({
  ...La,
  props: {
    modelValue: { default: () => [] },
    max: { default: 10 },
    renderTag: { type: Function, default: void 0 },
    closeable: { type: Boolean, default: !0 }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const n = e, a = X("dynamic-tag-"), s = I({
      get: () => n.modelValue,
      set: (r) => t("update:modelValue", r)
    }), l = U(null), o = F({
      newTag: "",
      typing: !1,
      // if user is typing
      clickAdd: () => {
        o.typing = !0, setTimeout(() => {
          var r;
          return (r = l.value) == null ? void 0 : r.focus();
        }, 0);
      },
      removeTag: (r) => {
        s.value = s.value.filter((i) => i !== r);
      },
      addTag: (r) => {
        const f = r.target.value.trim();
        f && s.value.push(f), o.deactive();
      },
      deactive: () => {
        o.newTag = "", o.typing = !1;
      }
    });
    return (r, i) => (u(), d("div", Oa, [
      e.renderTag !== void 0 ? (u(!0), d(S, { key: 0 }, M(w(s), (f, y) => (u(), E(ue(e.renderTag(f, y)), {
        key: y,
        class: "mr-2"
      }))), 128)) : (u(!0), d(S, { key: 1 }, M(w(s), (f) => (u(), E(Se, {
        key: f,
        closeable: e.closeable,
        class: "mr-2",
        onClose: (y) => o.removeTag(f)
      }, {
        default: B(() => [
          _(A(f), 1)
        ]),
        _: 2
      }, 1032, ["closeable", "onClose"]))), 128)),
      o.typing ? $(r.$slots, "input", {
        key: 2,
        submit: o.addTag,
        deactive: o.deactive
      }, () => [
        K(Se, null, {
          default: B(() => [
            c("div", Ma, [
              O(c("input", {
                id: w(a),
                ref_key: "newTagInputRef",
                ref: l,
                "onUpdate:modelValue": i[0] || (i[0] = (f) => o.newTag = f),
                class: "absolute inline h-full w-full border-none bg-transparent text-gray-500 outline-transparent focus:outline-none",
                onBlur: i[1] || (i[1] = //@ts-ignore
                (...f) => o.addTag && o.addTag(...f)),
                onKeyup: i[2] || (i[2] = rn((f) => r.$refs.newTagInputRef.blur(), ["enter"]))
              }, null, 40, Ua), [
                [un, o.newTag]
              ]),
              c("label", {
                for: w(a),
                class: "invisible",
                textContent: A(o.newTag)
              }, null, 8, za)
            ])
          ]),
          _: 1
        })
      ]) : k("", !0),
      w(s).length < e.max && !o.typing ? (u(), d("span", {
        key: 3,
        class: "cursor-pointer",
        onClick: i[3] || (i[3] = //@ts-ignore
        (...f) => o.clickAdd && o.clickAdd(...f))
      }, [
        $(r.$slots, "trigger", {}, () => [
          Ea
        ])
      ])) : k("", !0)
    ]));
  }
}), _a = (e) => ($t("data-v-0669237d"), e = e(), wt(), e), Ka = ["for"], Fa = {
  key: 0,
  class: "flex flex-wrap gap-1"
}, Na = {
  key: 1,
  class: "text-sm"
}, Ra = {
  key: 2,
  class: "text-gray-400"
}, Ga = /* @__PURE__ */ _a(() => /* @__PURE__ */ c("div", { class: "i-mdi:upload h-5 w-5 text-gray-400" }, null, -1)), Ha = ["id", "multiple"], qa = {
  name: "UFileInput"
}, Wa = /* @__PURE__ */ C({
  ...qa,
  props: {
    fileList: null,
    placeholder: { default: "Select a file" },
    multiple: { type: Boolean, default: !1 }
  },
  emits: ["change", "update:fileList"],
  setup(e, { emit: t }) {
    const n = e, a = X("file-input-"), s = I({
      get() {
        return n.fileList || [];
      },
      set(o) {
        t("update:fileList", o);
      }
    }), l = F({
      fileNames: [],
      onChange(o) {
        const r = o.target, i = Array.from(r.files || []);
        l.fileNames = i.map((f) => f.name), s.value = i.map((f) => ({
          id: X("file-"),
          name: f.name,
          file: f
        })), t("change", o);
      }
    });
    return (o, r) => (u(), d(S, null, [
      c("label", {
        for: w(a),
        class: p({ upload: !o.$slots.default })
      }, [
        o.$slots.default ? $(o.$slots, "default", { key: 0 }, void 0, !0) : (u(), d(S, { key: 1 }, [
          l.fileNames.length > 1 ? (u(), d("div", Fa, [
            (u(!0), d(S, null, M(l.fileNames, (i) => (u(), d("div", {
              key: i,
              class: "bg-primary-300 text-whit inline-block whitespace-nowrap rounded px-1 text-center text-xs"
            }, A(i), 1))), 128))
          ])) : l.fileNames.length === 1 ? (u(), d("div", Na, A(l.fileNames[0]), 1)) : (u(), d("div", Ra, A(e.placeholder), 1)),
          Ga
        ], 64))
      ], 10, Ka),
      c("input", P({
        id: w(a),
        class: "hidden",
        type: "file"
      }, o.$attrs, {
        multiple: e.multiple,
        onChange: r[0] || (r[0] = //@ts-ignore
        (...i) => l.onChange && l.onChange(...i))
      }), null, 16, Ha)
    ], 64));
  }
});
const Za = /* @__PURE__ */ ce(Wa, [["__scopeId", "data-v-0669237d"]]), Ya = { class: "mb-1 text-sm empty:hidden" }, Xa = {
  key: 2,
  class: "text-error-500"
}, Ja = { class: "flex" }, Qa = { class: "relative w-full flex items-center" }, el = {
  key: 0,
  class: "absolute left-3 z-10"
}, tl = ["type", "disabled", "placeholder"], nl = {
  key: 1,
  class: "absolute right-3 z-10"
}, al = {
  key: 0,
  class: "text-error-500 mt-1 text-xs"
}, ll = {
  name: "UInput",
  inheritAttrs: !1
}, ol = /* @__PURE__ */ C({
  ...ll,
  props: {
    modelValue: { default: "" },
    label: { default: "" },
    type: { default: "text" },
    size: { default: "md" },
    disabled: { type: Boolean, default: !1 },
    required: { type: Boolean, default: !1 },
    errorMessage: { default: "" },
    placeholder: { default: "Please input" },
    clearable: { type: Boolean, default: !1 },
    leftIcon: { default: "" },
    rightIcon: { default: "" }
  },
  emits: ["update:modelValue", "clear", "clickLeft", "clickRight"],
  setup(e, { emit: t }) {
    const n = e, a = I({
      get: () => n.modelValue,
      set: (l) => t("update:modelValue", l)
    }), s = F({
      focused: !1,
      clear() {
        a.value = "", t("clear");
      }
    });
    return (l, o) => (u(), d("div", {
      class: p(["w-full flex flex-col", { "opacity-60": e.disabled }])
    }, [
      c("label", Ya, [
        e.label ? (u(), d(S, { key: 0 }, [
          _(A(e.label), 1)
        ], 64)) : $(l.$slots, "default", { key: 1 }, void 0, !0),
        e.required ? (u(), d("span", Xa, "*")) : k("", !0)
      ]),
      c("div", Ja, [
        l.$slots.prepend ? (u(), d("span", {
          key: 0,
          class: p(["inline-flex items-center border border-r-0 rounded-l-md px-3 text-sm", {
            "py-0.5": e.size === "sm",
            "py-1.5": e.size === "md",
            "py-2": e.size === "lg"
          }])
        }, [
          $(l.$slots, "prepend", {}, void 0, !0)
        ], 2)) : k("", !0),
        c("div", Qa, [
          e.leftIcon ? (u(), d("div", el, [
            c("div", {
              class: p(["h-5 w-5", e.leftIcon]),
              onClick: o[0] || (o[0] = (r) => l.$emit("clickLeft"))
            }, null, 2)
          ])) : k("", !0),
          O(c("input", P({
            "onUpdate:modelValue": o[1] || (o[1] = (r) => Y(a) ? a.value = r : null)
          }, l.$attrs, {
            type: e.type,
            disabled: e.disabled,
            placeholder: e.placeholder,
            class: ["text-field-input focus:ring-primary-600", {
              // size
              "py-1": e.size === "sm",
              "py-2": e.size === "md",
              "py-3": e.size === "lg",
              // icon
              "pl-10": e.leftIcon,
              "pr-10": e.rightIcon && !e.clearable || !e.rightIcon && e.clearable,
              "pr-15": e.rightIcon && e.clearable,
              "px-3": !e.leftIcon && !e.rightIcon,
              prepend: l.$slots.prepend,
              append: l.$slots.append,
              danger: e.errorMessage,
              "cursor-not-allowed": e.disabled
            }],
            onFocus: o[2] || (o[2] = (r) => s.focused = !0),
            onBlur: o[3] || (o[3] = (r) => s.focused = !1)
          }), null, 16, tl), [
            [cn, w(a)]
          ]),
          e.rightIcon ? (u(), d("div", nl, [
            c("div", {
              class: p(["h-5 w-5", e.rightIcon]),
              onClick: o[4] || (o[4] = (r) => l.$emit("clickRight"))
            }, null, 2)
          ])) : k("", !0),
          c("div", {
            class: p(["absolute z-10", {
              "right-3": e.clearable && !e.rightIcon,
              "right-10": e.clearable && e.rightIcon
            }])
          }, [
            e.modelValue && e.clearable ? (u(), d("div", {
              key: 0,
              class: "i-mdi:close-circle-outline h-5 w-5 cursor-pointer text-gray-400 duration-300 hover:text-gray-500",
              onClick: o[5] || (o[5] = //@ts-ignore
              (...r) => s.clear && s.clear(...r))
            })) : k("", !0)
          ], 2)
        ]),
        l.$slots.append ? (u(), d("span", {
          key: 1,
          class: p(["inline-flex items-center border border-l-0 rounded-r-md px-3 text-sm", {
            "py-0.5": e.size === "sm",
            "py-1.5": e.size === "md",
            "py-2": e.size === "lg"
          }])
        }, [
          $(l.$slots, "append", {}, void 0, !0)
        ], 2)) : k("", !0)
      ]),
      e.errorMessage ? (u(), d("span", al, A(e.errorMessage), 1)) : k("", !0)
    ], 2));
  }
});
const sl = /* @__PURE__ */ ce(ol, [["__scopeId", "data-v-6653a735"]]), rl = /* @__PURE__ */ c("span", { class: "sr-only" }, "loading", -1), il = [
  rl
], ul = {
  name: "USpin"
}, Ae = /* @__PURE__ */ C({
  ...ul,
  props: {
    show: { type: Boolean, default: !0 },
    type: { default: "primary" },
    thickness: { default: 5 }
  },
  setup(e) {
    return (t, n) => (u(), d("div", P(t.$attrs, {
      class: {
        "h-8 w-8": !t.$attrs.hasOwnProperty("class")
      }
    }), [
      $(t.$slots, "icon", {}, () => [
        c("div", {
          class: p(["inline-block h-full w-full animate-spin border-gray-300 rounded-full bg-transparent", [`border-r-${e.type}-500`, `border-${e.thickness}`]])
        }, il, 2)
      ])
    ], 16));
  }
}), cl = {
  key: 1,
  class: "relative"
}, dl = { class: "absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2" }, fl = {
  name: "ULoading"
}, hl = /* @__PURE__ */ C({
  ...fl,
  props: {
    show: { type: Boolean, default: !0 }
  },
  setup(e) {
    return (t, n) => t.$slots.default ? (u(), d("div", cl, [
      c("div", {
        class: p({ "opacity-30": e.show })
      }, [
        $(t.$slots, "default")
      ], 2),
      c("div", dl, [
        e.show ? (u(), E(Ae, Ve(P({ key: 0 }, t.$attrs)), {
          icon: B(() => [
            $(t.$slots, "icon")
          ]),
          _: 3
        }, 16)) : k("", !0)
      ])
    ])) : (u(), d(S, { key: 0 }, [
      e.show ? (u(), E(Ae, Ve(P({ key: 0 }, t.$attrs)), dn({
        default: B(() => [
          $(t.$slots, "default")
        ]),
        _: 2
      }, [
        t.$slots.icon ? {
          name: "icon",
          fn: B(() => [
            $(t.$slots, "icon")
          ]),
          key: "0"
        } : void 0
      ]), 1040)) : k("", !0)
    ], 64));
  }
}), gl = ["onClick"], yl = { class: "min-h-full flex items-center justify-center p-2 sm:p-6" }, ml = /* @__PURE__ */ c("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  viewBox: "0 0 24 24",
  stroke: "currentColor"
}, [
  /* @__PURE__ */ c("path", {
    "stroke-linecap": "round",
    "stroke-linejoin": "round",
    "stroke-width": "2",
    d: "M6 18L18 6M6 6l12 12"
  })
], -1), vl = [
  ml
], pl = {
  key: 1,
  class: "mb-5 text-lg font-bold"
}, bl = {
  key: 2,
  class: "mt-5 sm:flex sm:flex-row-reverse sm:gap-2"
}, $l = {
  name: "UModal"
}, wl = /* @__PURE__ */ C({
  ...$l,
  props: {
    modelValue: { type: Boolean },
    dismissible: { type: Boolean, default: !0 },
    dismissButton: { type: Boolean, default: !0 },
    width: { default: "base" },
    padded: { type: Boolean, default: !0 },
    zIndex: { default: 30 }
  },
  emits: ["update:modelValue", "close"],
  setup(e, { emit: t }) {
    const n = e, a = I({
      get: () => n.modelValue,
      set: (l) => t("update:modelValue", l)
    });
    function s() {
      n.dismissible && (a.value = !1), t("close");
    }
    return (l, o) => (u(), E(pt, { to: "body" }, [
      c("div", {
        class: p(["fixed inset-0 overflow-y-auto transition-all ease-in", [
          w(a) ? "visible" : "invisible duration-100 ease-in"
        ]]),
        style: ve({
          zIndex: e.zIndex
        })
      }, [
        c("div", {
          class: p(["fixed inset-0 bg-gray-500 transition-opacity dark:bg-gray-600", [
            w(a) ? "opacity-75 duration-75 ease-out" : "opacity-0 duration-75 ease-in"
          ]]),
          onClick: D(s, ["self"])
        }, null, 10, gl),
        c("div", yl, [
          c("div", P(l.$attrs, {
            class: ["relative inline-block w-full rounded-lg bg-white shadow-xl transition-all dark:bg-gray-900", [
              e.padded ? "p-4 sm:p-6" : "p2",
              {
                "sm:max-w-sm": e.width === "sm",
                "sm:max-w-lg": e.width === "base",
                "sm:max-w-xl": e.width === "md",
                "sm:max-w-3xl": e.width === "lg",
                "sm:max-w-5xl": e.width === "xl",
                "sm:max-w-full": e.width === "full"
              },
              w(a) ? "translate-y-0 opacity-100 duration-300 sm:scale-100" : "translate-y-4 opacity-0 duration-300 sm:translate-y-0 sm:scale-95"
            ]],
            role: "dialog",
            "aria-modal": "true"
          }), [
            e.dismissButton ? (u(), d("button", {
              key: 0,
              class: "absolute right-4 top-4 h-6 w-6 rounded-full bg-gray-100 p-1 text-gray-700 hover:bg-gray-200 focus:outline-none focus:ring-2 focus:ring-gray-500",
              "aria-label": "close",
              onClick: s
            }, vl)) : k("", !0),
            l.$slots.header ? (u(), d("div", pl, [
              $(l.$slots, "header", {}, () => [
                _(" Header ")
              ])
            ])) : k("", !0),
            $(l.$slots, "default"),
            l.$slots.footer ? (u(), d("div", bl, [
              $(l.$slots, "footer", {}, () => [
                _(" Footer ")
              ])
            ])) : k("", !0)
          ], 16)
        ])
      ], 6)
    ]));
  }
});
const xl = { class: "relative inline-block text-left" }, kl = {
  name: "UPopover"
}, Tl = /* @__PURE__ */ C({
  ...kl,
  props: {
    position: { default: "auto" },
    trigger: { default: "click" },
    dissmissable: { type: Boolean, default: !1 },
    sustain: { type: Boolean, default: !1 },
    hoverAlive: { type: Boolean, default: !1 }
  },
  emits: ["open", "close"],
  setup(e, { emit: t }) {
    const n = e, a = U(null), s = U(null), l = F({
      inPanel: !1,
      status: !1,
      onClick: () => {
        n.trigger === "click" && l.toggleStatus();
      },
      onMouseenter() {
        n.trigger === "hover" && l.open();
      },
      onMouseleave() {
        n.trigger === "hover" && l.close();
      },
      toggleStatus() {
        l.status = !l.status, l.status ? l.open() : l.close();
      },
      open() {
        l.status = !0, t("open"), bt(() => {
          const o = a.value.getBoundingClientRect();
          s.value.style.left = `${o.x}px`, s.value.style.top = `${o.y}px`;
          let r, i;
          if (n.position === "auto") {
            const f = window.innerHeight / 2;
            r = `calc(${o.width / 2}px - 50%)`, i = o.top > f ? "calc(-100% - 0.25rem)" : `calc(${o.height}px + 0.25rem)`;
          } else
            n.position === "left" ? (r = "calc(-100% - 0.25rem)", i = `calc(${o.height / 2}px - 50%)`) : n.position === "right" ? (r = `calc(${o.width}px + 0.25rem)`, i = `calc(${o.height / 2}px - 50%)`) : n.position === "top" ? (r = `calc(${o.width / 2}px - 50%)`, i = "calc(-100% - 0.25rem)") : n.position === "bottom" && (r = `calc(${o.width / 2}px - 50%)`, i = `calc(${o.height}px + 0.25rem)`);
          s.value.style.transform = `translate(${r}, ${i})`;
        });
      },
      close() {
        l.status && t("close"), setTimeout(() => l.status = !1, 100);
      }
    });
    return (o, r) => O((u(), d("div", xl, [
      c("div", {
        ref_key: "target",
        ref: a,
        class: "w-full inline-flex items-center justify-center rounded-md",
        onClick: r[0] || (r[0] = //@ts-ignore
        (...i) => l.onClick && l.onClick(...i)),
        onMouseenter: r[1] || (r[1] = //@ts-ignore
        (...i) => l.onMouseenter && l.onMouseenter(...i)),
        onMouseleave: r[2] || (r[2] = (i) => e.sustain ? null : l.onMouseleave())
      }, [
        $(o.$slots, "default")
      ], 544),
      K(pe, {
        "enter-active-class": "transition-opacity duration-300 ease-out",
        "enter-from-class": "opacity-0",
        "enter-to-class": "opacity-100",
        "leave-active-class": "duration-200 ease-in",
        "leave-from-class": "opacity-100",
        "leave-to-class": "opacity-0"
      }, {
        default: B(() => [
          e.hoverAlive && l.inPanel || l.status ? (u(), d("div", {
            key: 0,
            ref_key: "dropdown",
            ref: s,
            class: "fixed left-0 top-0 z-10 min-w-max rounded-lg bg-white shadow-lg dark:bg-slate-800",
            tabindex: "-1"
          }, [
            c("div", {
              class: "py-1",
              onMouseenter: r[3] || (r[3] = (i) => l.inPanel = !0),
              onMouseleave: r[4] || (r[4] = (i) => l.inPanel = !1),
              onClick: r[5] || (r[5] = (i) => e.dissmissable ? l.close() : null)
            }, [
              $(o.$slots, "content")
            ], 32)
          ], 512)) : k("", !0)
        ]),
        _: 3
      })
    ])), [
      [w(ze), l.close]
    ]);
  }
}), Cl = {
  key: 0,
  class: "overflow-hidden rounded-lg bg-gray-200 dark:bg-gray-700"
}, Vl = {
  key: 1,
  class: "-rotate-90",
  viewBox: "0 0 100 100"
}, jl = /* @__PURE__ */ c("circle", {
  class: "text-gray-200 dark:text-gray-700",
  stroke: "currentColor",
  "stroke-linejoin": "round",
  "stroke-linecap": "round",
  "stroke-width": "8",
  fill: "none",
  cx: "50",
  cy: "50",
  r: "40"
}, null, -1), Sl = {
  key: 2,
  class: "inline-block space-x-1"
}, Al = {
  name: "UProgress"
}, Il = /* @__PURE__ */ C({
  ...Al,
  props: {
    type: { default: "primary" },
    percentage: null,
    scoreLength: { default: 5 },
    tag: { default: "div" },
    variant: { default: "bar" }
  },
  setup(e) {
    const t = e, n = I(() => 251 / 100 * t.percentage);
    function a(s) {
      return (s - 1) * (100 / t.scoreLength) <= t.percentage;
    }
    return (s, l) => (u(), E(ue(e.tag), null, {
      default: B(() => [
        e.variant === "bar" ? (u(), d("div", Cl, [
          c("div", P(s.$attrs, {
            class: ["h-2 rounded-lg", `bg-${e.type}-500 text-${e.type}-500`],
            style: { width: `${e.percentage}%` }
          }), null, 16)
        ])) : e.variant === "circle" ? (u(), d("svg", Vl, [
          jl,
          c("circle", {
            class: p(`bg-${e.type}-500 text-${e.type}-500`),
            style: ve({ strokeDasharray: `${w(n)} 252` }),
            stroke: "currentColor",
            "stroke-linejoin": "round",
            "stroke-linecap": "round",
            "stroke-width": "8",
            fill: "none",
            cx: "50",
            cy: "50",
            r: "40"
          }, null, 6)
        ])) : e.variant === "score" ? (u(), d("div", Sl, [
          (u(!0), d(S, null, M(e.scoreLength, (o) => (u(), d("div", {
            key: o,
            class: p(["inline-block h-2 w-3 rounded-sm", {
              "bg-gray-200": !a(o),
              [`bg-${e.type}-500`]: a(o)
            }])
          }, null, 2))), 128))
        ])) : k("", !0)
      ]),
      _: 1
    }));
  }
}), Bl = ["for"], Pl = ["id", "value", "name", "disabled"], Ol = {
  name: "URadio"
}, Ml = /* @__PURE__ */ C({
  ...Ol,
  props: {
    modelValue: { default: void 0 },
    type: { default: "primary" },
    value: { default: void 0 },
    disabled: { type: Boolean, default: !1 },
    bordered: { type: Boolean, default: !1 },
    hidden: { type: Boolean, default: !1 },
    circleColored: { type: Boolean, default: !1 }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const n = e, a = X("radio-"), s = I({
      get: () => n.modelValue,
      set: (o) => t("update:modelValue", o)
    }), l = I(() => n.value === s.value);
    return (o, r) => (u(), d("label", {
      for: w(a),
      class: p(["inline-flex select-none items-center font-medium leading-6 text-gray-900", [
        e.disabled ? "cursor-not-allowed opacity-60" : "cursor-pointer",
        {
          "border-1 rounded p-2 transition-colors duration-150 ease-in-out": e.bordered,
          "border-gray-300": e.bordered && !w(l) && !e.disabled,
          [`border-${e.type}-500`]: e.bordered && w(l)
        }
      ]])
    }, [
      O(c("input", P({ id: w(a) }, o.$attrs, {
        "onUpdate:modelValue": r[0] || (r[0] = (i) => Y(s) ? s.value = i : null),
        type: "radio",
        value: e.value,
        name: w(a),
        disabled: e.disabled,
        class: [
          "h-4 w-4",
          [
            e.circleColored ? `border-${e.type}-500` : "border-gray-300",
            e.disabled ? "cursor-not-allowed opacity-60" : "cursor-pointer",
            `text-${e.type}-600 focus:ring-${e.type}-500`,
            {
              "hidden sr-only": e.hidden
            }
          ]
        ]
      }), null, 16, Pl), [
        [fn, w(s)]
      ]),
      c("div", {
        class: p({ "pl-2": o.$slots.default && !e.hidden })
      }, [
        $(o.$slots, "default", { checked: w(l) })
      ], 2)
    ], 10, Bl));
  }
}), Ul = {
  id: "listbox-label",
  class: "block text-sm font-medium leading-6 text-gray-900"
}, zl = { class: "relative" }, El = ["disabled", "placeholder", "readonly", "value"], Ll = {
  key: 1,
  class: "i-mdi:unfold-more-horizontal h-5 w-5"
}, Dl = {
  key: 0,
  class: "absolute z-10 mt-1 max-h-60 w-full overflow-auto rounded-md bg-white py-1 text-sm shadow-lg ring-1 ring-black ring-opacity-5 focus:outline-none",
  tabindex: "-1"
}, _l = ["onClick"], Kl = { class: "block truncate font-normal" }, Fl = {
  key: 0,
  class: "absolute inset-y-0 right-0 flex items-center pr-3 text-white"
}, Nl = /* @__PURE__ */ c("svg", {
  class: "h-5 w-5",
  viewBox: "0 0 20 20",
  fill: "currentColor",
  "aria-hidden": "true"
}, [
  /* @__PURE__ */ c("path", {
    "fill-rule": "evenodd",
    d: "M16.704 4.153a.75.75 0 01.143 1.052l-8 10.5a.75.75 0 01-1.127.075l-4.5-4.5a.75.75 0 011.06-1.06l3.894 3.893 7.48-9.817a.75.75 0 011.05-.143z",
    "clip-rule": "evenodd"
  })
], -1), Rl = [
  Nl
], Gl = {
  key: 0,
  class: "text-xs text-red-500"
}, Hl = {
  name: "USelect"
}, ql = /* @__PURE__ */ C({
  ...Hl,
  props: {
    label: { default: "" },
    modelValue: { default: null },
    placeholder: { default: "Please select" },
    disabled: { type: Boolean, default: !1 },
    clearable: { type: Boolean, default: !1 },
    filterable: { type: Boolean, default: !1 },
    options: { default: () => [] },
    errorMessage: { default: "" }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const n = e, a = I({
      get: () => n.modelValue,
      set: (r) => {
        o.inputValue = s(r), t("update:modelValue", r);
      }
    });
    q(() => n.modelValue, (r) => {
      setTimeout(() => o.inputValue = s(r), 50);
    });
    function s(r) {
      var i, f;
      return ((f = (i = n.options) == null ? void 0 : i.find((y) => y.value === r)) == null ? void 0 : f.label) || "";
    }
    const l = I(() => {
      var r;
      return n.filterable ? (r = n.options) == null ? void 0 : r.filter((i) => {
        var f;
        return (f = i.label) == null ? void 0 : f.includes(o.inputValue);
      }) : n.options;
    }), o = F({
      show: !1,
      inputValue: s(n.modelValue),
      onSelect(r) {
        a.value = r, o.show = !1;
      },
      filter(r) {
        n.filterable && (o.inputValue = r, o.show = !0);
      },
      toggleMenu() {
        n.disabled || (o.show = !o.show);
      }
    });
    return (r, i) => (u(), d("div", null, [
      c("label", Ul, A(e.label), 1),
      O((u(), d("div", zl, [
        c("input", {
          type: "text",
          class: p(["focus:ring-primary-600 w-full border-0 rounded-md bg-white py-1.5 pl-3 pr-8 text-sm leading-6 shadow-sm ring-1 ring-inset placeholder:text-slate-400 focus:ring-2 focus:ring-inset", [
            e.disabled ? "cursor-not-allowed opacity-40" : "cursor-default",
            e.errorMessage ? "text-red-500 ring-red-500 focus:ring-red-500" : "text-gray-700 ring-gray-300 focus:ring-primary-600"
          ]]),
          disabled: e.disabled,
          placeholder: e.placeholder,
          readonly: !e.filterable,
          value: o.inputValue,
          onClick: i[0] || (i[0] = () => o.toggleMenu()),
          onInput: i[1] || (i[1] = (f) => {
            o.show = !0, o.inputValue = f.target.value;
          })
        }, null, 42, El),
        c("button", {
          class: p(["group absolute inset-y-0 right-0 mr-1.5 text-gray-400", [e.disabled ? "cursor-not-allowed opacity-40" : "cursor-default"]]),
          onClick: i[3] || (i[3] = //@ts-ignore
          (...f) => o.toggleMenu && o.toggleMenu(...f))
        }, [
          e.clearable && w(a) ? (u(), d("div", {
            key: 0,
            class: "i-mdi:unfold-more-horizontal group-hover:i-mdi:close-circle-outline h-5 w-5 text-gray-400 duration-100 group-hover:h-5 group-hover:w-5 hover:cursor-pointer",
            onClick: i[2] || (i[2] = D((f) => a.value = null, ["stop"]))
          })) : (u(), d("div", Ll))
        ], 2),
        K(pe, {
          "leave-active-class": "transition ease-in duration-100",
          "leave-from-class": "opacity-100",
          "leave-to-class": "opacity-0"
        }, {
          default: B(() => [
            o.show && w(l).length ? (u(), d("ul", Dl, [
              (u(!0), d(S, null, M(w(l), (f) => (u(), d("li", {
                id: "listbox-option-0",
                key: f.value,
                class: p(["text relative cursor-default select-none py-2 pl-3 duration-100", {
                  "bg-primary-500 text-white ": e.modelValue === f.value,
                  "hover:bg-primary-500 text-gray-900 hover:text-white": e.modelValue !== f.value
                }]),
                onClick: (y) => o.onSelect(f.value)
              }, [
                $(r.$slots, "option", {
                  item: f,
                  active: e.modelValue === f.value
                }, () => [
                  c("span", Kl, A(f.label), 1),
                  e.modelValue === f.value ? (u(), d("span", Fl, Rl)) : k("", !0)
                ])
              ], 10, _l))), 128))
            ])) : k("", !0)
          ]),
          _: 3
        })
      ])), [
        [w(ze), () => o.show = !1]
      ]),
      e.errorMessage ? (u(), d("div", Gl, A(e.errorMessage), 1)) : k("", !0)
    ]));
  }
}), Wl = {
  name: "USkeleton",
  inheritAttrs: !1
}, Zl = /* @__PURE__ */ C({
  ...Wl,
  props: {
    as: { default: "div" }
  },
  setup(e) {
    return (t, n) => {
      var a;
      return u(), E(ue(e.as), P(t.$attrs, {
        class: ["animate-pulse", {
          "bg-gray-200": !((a = t.$attrs.class) != null && a.includes("bg-"))
        }]
      }), {
        default: B(() => [
          _("  ")
        ]),
        _: 1
      }, 16, ["class"]);
    };
  }
}), Yl = { class: "flex items-center" }, Xl = ["for"], Jl = { class: "mr-2 font-medium leading-6 text-gray-900 empty:hidden" }, Ql = {
  tabindex: "0",
  class: "relative"
}, eo = ["id", "disabled"], to = /* @__PURE__ */ xt('<defs><linearGradient id="mingcuteLoadingLine0" x1="50%" x2="50%" y1="5.271%" y2="91.793%"><stop offset="0%" stop-color="#888888"></stop><stop offset="100%" stop-color="currentColor" stop-opacity=".55"></stop></linearGradient><linearGradient id="mingcuteLoadingLine1" x1="50%" x2="50%" y1="8.877%" y2="90.415%"><stop offset="0%" stop-color="currentColor" stop-opacity="0"></stop><stop offset="100%" stop-color="currentColor" stop-opacity=".55"></stop></linearGradient></defs><g fill="none"><path d="M24 0v24H0V0h24ZM12.593 23.258l-.011.002l-.071.035l-.02.004l-.014-.004l-.071-.035c-.01-.004-.019-.001-.024.005l-.004.01l-.017.428l.005.02l.01.013l.104.074l.015.004l.012-.004l.104-.074l.012-.016l.004-.017l-.017-.427c-.002-.01-.009-.017-.017-.018Zm.265-.113l-.013.002l-.185.093l-.01.01l-.003.011l.018.43l.005.012l.008.007l.201.093c.012.004.023 0 .029-.008l.004-.014l-.034-.614c-.003-.012-.01-.02-.02-.022Zm-.715.002a.023.023 0 0 0-.027.006l-.006.014l-.034.614c0 .012.007.02.017.024l.015-.002l.201-.093l.01-.008l.004-.011l.017-.43l-.003-.012l-.01-.01l-.184-.092Z"></path><path fill="url(#mingcuteLoadingLine0)" d="M8.886.006a1 1 0 0 1 .22 1.988A8.001 8.001 0 0 0 10 17.944v2c-5.523 0-10-4.476-10-10C0 4.838 3.848.566 8.886.007Z" transform="translate(2 2.055)"></path><path fill="url(#mingcuteLoadingLine1)" d="M14.322 1.985a1 1 0 0 1 1.392-.248A9.988 9.988 0 0 1 20 9.945c0 5.523-4.477 10-10 10v-2a8 8 0 0 0 4.57-14.567a1 1 0 0 1-.248-1.393Z" transform="translate(2 2.055)"></path></g>', 2), no = [
  to
], ao = {
  key: 0,
  d: "M3.707 5.293a1 1 0 00-1.414 1.414l1.414-1.414zM5 8l-.707.707a1 1 0 001.414 0L5 8zm4.707-3.293a1 1 0 00-1.414-1.414l1.414 1.414zm-7.414 2l2 2 1.414-1.414-2-2-1.414 1.414zm3.414 2l4-4-1.414-1.414-4 4 1.414 1.414z"
}, lo = {
  key: 1,
  d: "M4 8l2-2m0 0l2-2M6 6L4 4m2 2l2 2",
  stroke: "currentColor",
  "stroke-width": "2",
  "stroke-linecap": "round",
  "stroke-linejoin": "round"
}, oo = {
  name: "USwitch",
  inheritAttrs: !1
}, so = /* @__PURE__ */ C({
  ...oo,
  props: {
    modelValue: { type: [Boolean, String, Number], default: !1 },
    type: { default: "primary" },
    size: { default: "base" },
    loading: { type: Boolean, default: !1 },
    checked: { type: Boolean },
    disabled: { type: Boolean, default: !1 },
    icon: { type: Boolean, default: !1 },
    readonly: { type: Boolean, default: !1 },
    checkedValue: { default: !0 },
    uncheckedValue: { default: !1 }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const n = e, a = X("switch-"), s = I({
      get: () => n.checkedValue && n.modelValue === n.checkedValue ? !0 : n.uncheckedValue && n.modelValue === n.uncheckedValue ? !1 : n.modelValue,
      set: (o) => {
        n.checkedValue && o ? o = n.checkedValue : n.uncheckedValue && !o && (o = n.uncheckedValue), t("update:modelValue", o);
      }
    }), l = {
      success: {
        color: "bg-success-500",
        text: "text-success-500",
        focus: "focus:ring-success-500"
      },
      info: {
        color: "bg-info-500",
        text: "text-info-500",
        focus: "focus:ring-info-500"
      },
      warning: {
        color: "bg-warning-500",
        text: "text-warning-500",
        focus: "focus:ring-warning-500"
      },
      error: {
        color: "bg-error-500",
        text: "text-error-500",
        focus: "focus:ring-error-500"
      },
      primary: {
        color: "bg-primary-500",
        text: "text-primary-500",
        focus: "focus:ring-primary-500"
      },
      secondary: {
        color: "bg-secondary-500",
        text: "text-secondary-500",
        focus: "focus:ring-secondary-500"
      },
      accent: {
        color: "bg-accent-500",
        text: "text-accent-500",
        focus: "focus:ring-accent-500"
      }
    };
    return (o, r) => (u(), d("div", Yl, [
      c("label", {
        for: w(a),
        class: p(["flex select-none items-center", [e.disabled ? "cursor-not-allowed opacity-60" : e.readonly ? "" : "cursor-pointer"]])
      }, [
        c("div", Jl, [
          $(o.$slots, "default")
        ]),
        c("div", Ql, [
          O(c("input", P({
            id: w(a),
            "onUpdate:modelValue": r[0] || (r[0] = (i) => Y(s) ? s.value = i : null)
          }, o.$attrs, {
            type: "checkbox",
            disabled: e.disabled,
            class: "sr-only"
          }), null, 16, eo), [
            [se, w(s)]
          ]),
          c("div", {
            class: p(["block rounded-full", [
              w(s) || e.checked ? l[e.type].color : "bg-gray-200",
              {
                "h-4 w-9": e.size === "sm",
                "h-6 w-11": e.size === "base",
                "h-9 w-14": e.size === "lg"
              }
            ]])
          }, null, 2),
          c("div", {
            class: p(["absolute left-0 top-0.5 rounded-full bg-white transition", [
              w(s) || e.checked ? "translate-x-5" : "translate-x-0",
              {
                "h-3 w-3": e.size === "sm",
                "h-5 w-5": e.size === "base",
                "h-8 w-8": e.size === "lg"
              }
            ]])
          }, [
            e.loading ? (u(), d("svg", {
              key: 0,
              class: p(["animate-spin", [
                {
                  "h-3 w-3": e.size === "sm",
                  "h-5 w-5": e.size === "base",
                  "h-8 w-8": e.size === "lg"
                }
              ]]),
              xmlns: "http://www.w3.org/2000/svg",
              width: "32",
              height: "32",
              viewBox: "0 0 24 24"
            }, no, 2)) : e.icon ? (u(), d("svg", {
              key: 1,
              fill: "currentColor",
              viewBox: "0 0 12 12",
              class: p([
                e.checked || w(s) ? l[e.type].text : "text-gray-200",
                {
                  "h-3 w-3": e.size === "sm",
                  "h-5 w-5": e.size === "base",
                  "h-8 w-8": e.size === "lg"
                }
              ])
            }, [
              e.checked || w(s) ? (u(), d("path", ao)) : (u(), d("path", lo))
            ], 2)) : k("", !0)
          ], 2)
        ])
      ], 10, Xl)
    ]));
  }
});
function ro() {
  this.__data__ = [], this.size = 0;
}
function Pt(e, t) {
  return e === t || e !== e && t !== t;
}
function $e(e, t) {
  for (var n = e.length; n--; )
    if (Pt(e[n][0], t))
      return n;
  return -1;
}
var io = Array.prototype, uo = io.splice;
function co(e) {
  var t = this.__data__, n = $e(t, e);
  if (n < 0)
    return !1;
  var a = t.length - 1;
  return n == a ? t.pop() : uo.call(t, n, 1), --this.size, !0;
}
function fo(e) {
  var t = this.__data__, n = $e(t, e);
  return n < 0 ? void 0 : t[n][1];
}
function ho(e) {
  return $e(this.__data__, e) > -1;
}
function go(e, t) {
  var n = this.__data__, a = $e(n, e);
  return a < 0 ? (++this.size, n.push([e, t])) : n[a][1] = t, this;
}
function N(e) {
  var t = -1, n = e == null ? 0 : e.length;
  for (this.clear(); ++t < n; ) {
    var a = e[t];
    this.set(a[0], a[1]);
  }
}
N.prototype.clear = ro;
N.prototype.delete = co;
N.prototype.get = fo;
N.prototype.has = ho;
N.prototype.set = go;
function yo() {
  this.__data__ = new N(), this.size = 0;
}
function mo(e) {
  var t = this.__data__, n = t.delete(e);
  return this.size = t.size, n;
}
function vo(e) {
  return this.__data__.get(e);
}
function po(e) {
  return this.__data__.has(e);
}
function de(e) {
  var t = typeof e;
  return e != null && (t == "object" || t == "function");
}
var bo = "[object AsyncFunction]", $o = "[object Function]", wo = "[object GeneratorFunction]", xo = "[object Proxy]";
function Ot(e) {
  if (!de(e))
    return !1;
  var t = Q(e);
  return t == $o || t == wo || t == bo || t == xo;
}
var ko = L["__core-js_shared__"];
const Te = ko;
var Xe = function() {
  var e = /[^.]+$/.exec(Te && Te.keys && Te.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function To(e) {
  return !!Xe && Xe in e;
}
var Co = Function.prototype, Vo = Co.toString;
function W(e) {
  if (e != null) {
    try {
      return Vo.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var jo = /[\\^$.*+?()[\]{}|]/g, So = /^\[object .+?Constructor\]$/, Ao = Function.prototype, Io = Object.prototype, Bo = Ao.toString, Po = Io.hasOwnProperty, Oo = RegExp(
  "^" + Bo.call(Po).replace(jo, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function Mo(e) {
  if (!de(e) || To(e))
    return !1;
  var t = Ot(e) ? Oo : So;
  return t.test(W(e));
}
function Uo(e, t) {
  return e == null ? void 0 : e[t];
}
function Z(e, t) {
  var n = Uo(e, t);
  return Mo(n) ? n : void 0;
}
var zo = Z(L, "Map");
const re = zo;
var Eo = Z(Object, "create");
const ie = Eo;
function Lo() {
  this.__data__ = ie ? ie(null) : {}, this.size = 0;
}
function Do(e) {
  var t = this.has(e) && delete this.__data__[e];
  return this.size -= t ? 1 : 0, t;
}
var _o = "__lodash_hash_undefined__", Ko = Object.prototype, Fo = Ko.hasOwnProperty;
function No(e) {
  var t = this.__data__;
  if (ie) {
    var n = t[e];
    return n === _o ? void 0 : n;
  }
  return Fo.call(t, e) ? t[e] : void 0;
}
var Ro = Object.prototype, Go = Ro.hasOwnProperty;
function Ho(e) {
  var t = this.__data__;
  return ie ? t[e] !== void 0 : Go.call(t, e);
}
var qo = "__lodash_hash_undefined__";
function Wo(e, t) {
  var n = this.__data__;
  return this.size += this.has(e) ? 0 : 1, n[e] = ie && t === void 0 ? qo : t, this;
}
function H(e) {
  var t = -1, n = e == null ? 0 : e.length;
  for (this.clear(); ++t < n; ) {
    var a = e[t];
    this.set(a[0], a[1]);
  }
}
H.prototype.clear = Lo;
H.prototype.delete = Do;
H.prototype.get = No;
H.prototype.has = Ho;
H.prototype.set = Wo;
function Zo() {
  this.size = 0, this.__data__ = {
    hash: new H(),
    map: new (re || N)(),
    string: new H()
  };
}
function Yo(e) {
  var t = typeof e;
  return t == "string" || t == "number" || t == "symbol" || t == "boolean" ? e !== "__proto__" : e === null;
}
function we(e, t) {
  var n = e.__data__;
  return Yo(t) ? n[typeof t == "string" ? "string" : "hash"] : n.map;
}
function Xo(e) {
  var t = we(this, e).delete(e);
  return this.size -= t ? 1 : 0, t;
}
function Jo(e) {
  return we(this, e).get(e);
}
function Qo(e) {
  return we(this, e).has(e);
}
function es(e, t) {
  var n = we(this, e), a = n.size;
  return n.set(e, t), this.size += n.size == a ? 0 : 1, this;
}
function te(e) {
  var t = -1, n = e == null ? 0 : e.length;
  for (this.clear(); ++t < n; ) {
    var a = e[t];
    this.set(a[0], a[1]);
  }
}
te.prototype.clear = Zo;
te.prototype.delete = Xo;
te.prototype.get = Jo;
te.prototype.has = Qo;
te.prototype.set = es;
var ts = 200;
function ns(e, t) {
  var n = this.__data__;
  if (n instanceof N) {
    var a = n.__data__;
    if (!re || a.length < ts - 1)
      return a.push([e, t]), this.size = ++n.size, this;
    n = this.__data__ = new te(a);
  }
  return n.set(e, t), this.size = n.size, this;
}
function ne(e) {
  var t = this.__data__ = new N(e);
  this.size = t.size;
}
ne.prototype.clear = yo;
ne.prototype.delete = mo;
ne.prototype.get = vo;
ne.prototype.has = po;
ne.prototype.set = ns;
function as(e, t) {
  for (var n = -1, a = e == null ? 0 : e.length; ++n < a && t(e[n], n, e) !== !1; )
    ;
  return e;
}
var ls = function() {
  try {
    var e = Z(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}();
const Je = ls;
function Mt(e, t, n) {
  t == "__proto__" && Je ? Je(e, t, {
    configurable: !0,
    enumerable: !0,
    value: n,
    writable: !0
  }) : e[t] = n;
}
var os = Object.prototype, ss = os.hasOwnProperty;
function Ut(e, t, n) {
  var a = e[t];
  (!(ss.call(e, t) && Pt(a, n)) || n === void 0 && !(t in e)) && Mt(e, t, n);
}
function xe(e, t, n, a) {
  var s = !n;
  n || (n = {});
  for (var l = -1, o = t.length; ++l < o; ) {
    var r = t[l], i = a ? a(n[r], e[r], r, n, e) : void 0;
    i === void 0 && (i = e[r]), s ? Mt(n, r, i) : Ut(n, r, i);
  }
  return n;
}
function rs(e, t) {
  for (var n = -1, a = Array(e); ++n < e; )
    a[n] = t(n);
  return a;
}
var is = "[object Arguments]";
function Qe(e) {
  return ee(e) && Q(e) == is;
}
var zt = Object.prototype, us = zt.hasOwnProperty, cs = zt.propertyIsEnumerable, ds = Qe(function() {
  return arguments;
}()) ? Qe : function(e) {
  return ee(e) && us.call(e, "callee") && !cs.call(e, "callee");
};
const fs = ds;
function hs() {
  return !1;
}
var Et = typeof exports == "object" && exports && !exports.nodeType && exports, et = Et && typeof module == "object" && module && !module.nodeType && module, gs = et && et.exports === Et, tt = gs ? L.Buffer : void 0, ys = tt ? tt.isBuffer : void 0, ms = ys || hs;
const Lt = ms;
var vs = 9007199254740991, ps = /^(?:0|[1-9]\d*)$/;
function bs(e, t) {
  var n = typeof e;
  return t = t ?? vs, !!t && (n == "number" || n != "symbol" && ps.test(e)) && e > -1 && e % 1 == 0 && e < t;
}
var $s = 9007199254740991;
function Dt(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= $s;
}
var ws = "[object Arguments]", xs = "[object Array]", ks = "[object Boolean]", Ts = "[object Date]", Cs = "[object Error]", Vs = "[object Function]", js = "[object Map]", Ss = "[object Number]", As = "[object Object]", Is = "[object RegExp]", Bs = "[object Set]", Ps = "[object String]", Os = "[object WeakMap]", Ms = "[object ArrayBuffer]", Us = "[object DataView]", zs = "[object Float32Array]", Es = "[object Float64Array]", Ls = "[object Int8Array]", Ds = "[object Int16Array]", _s = "[object Int32Array]", Ks = "[object Uint8Array]", Fs = "[object Uint8ClampedArray]", Ns = "[object Uint16Array]", Rs = "[object Uint32Array]", V = {};
V[zs] = V[Es] = V[Ls] = V[Ds] = V[_s] = V[Ks] = V[Fs] = V[Ns] = V[Rs] = !0;
V[ws] = V[xs] = V[Ms] = V[ks] = V[Us] = V[Ts] = V[Cs] = V[Vs] = V[js] = V[Ss] = V[As] = V[Is] = V[Bs] = V[Ps] = V[Os] = !1;
function Gs(e) {
  return ee(e) && Dt(e.length) && !!V[Q(e)];
}
function Ee(e) {
  return function(t) {
    return e(t);
  };
}
var _t = typeof exports == "object" && exports && !exports.nodeType && exports, oe = _t && typeof module == "object" && module && !module.nodeType && module, Hs = oe && oe.exports === _t, Ce = Hs && Ct.process, qs = function() {
  try {
    var e = oe && oe.require && oe.require("util").types;
    return e || Ce && Ce.binding && Ce.binding("util");
  } catch {
  }
}();
const J = qs;
var nt = J && J.isTypedArray, Ws = nt ? Ee(nt) : Gs;
const Zs = Ws;
var Ys = Object.prototype, Xs = Ys.hasOwnProperty;
function Kt(e, t) {
  var n = be(e), a = !n && fs(e), s = !n && !a && Lt(e), l = !n && !a && !s && Zs(e), o = n || a || s || l, r = o ? rs(e.length, String) : [], i = r.length;
  for (var f in e)
    (t || Xs.call(e, f)) && !(o && // Safari 9 has enumerable `arguments.length` in strict mode.
    (f == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
    s && (f == "offset" || f == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
    l && (f == "buffer" || f == "byteLength" || f == "byteOffset") || // Skip index properties.
    bs(f, i))) && r.push(f);
  return r;
}
var Js = Object.prototype;
function Le(e) {
  var t = e && e.constructor, n = typeof t == "function" && t.prototype || Js;
  return e === n;
}
function Ft(e, t) {
  return function(n) {
    return e(t(n));
  };
}
var Qs = Ft(Object.keys, Object);
const er = Qs;
var tr = Object.prototype, nr = tr.hasOwnProperty;
function ar(e) {
  if (!Le(e))
    return er(e);
  var t = [];
  for (var n in Object(e))
    nr.call(e, n) && n != "constructor" && t.push(n);
  return t;
}
function Nt(e) {
  return e != null && Dt(e.length) && !Ot(e);
}
function De(e) {
  return Nt(e) ? Kt(e) : ar(e);
}
function lr(e, t) {
  return e && xe(t, De(t), e);
}
function or(e) {
  var t = [];
  if (e != null)
    for (var n in Object(e))
      t.push(n);
  return t;
}
var sr = Object.prototype, rr = sr.hasOwnProperty;
function ir(e) {
  if (!de(e))
    return or(e);
  var t = Le(e), n = [];
  for (var a in e)
    a == "constructor" && (t || !rr.call(e, a)) || n.push(a);
  return n;
}
function _e(e) {
  return Nt(e) ? Kt(e, !0) : ir(e);
}
function ur(e, t) {
  return e && xe(t, _e(t), e);
}
var Rt = typeof exports == "object" && exports && !exports.nodeType && exports, at = Rt && typeof module == "object" && module && !module.nodeType && module, cr = at && at.exports === Rt, lt = cr ? L.Buffer : void 0, ot = lt ? lt.allocUnsafe : void 0;
function dr(e, t) {
  if (t)
    return e.slice();
  var n = e.length, a = ot ? ot(n) : new e.constructor(n);
  return e.copy(a), a;
}
function fr(e, t) {
  var n = -1, a = e.length;
  for (t || (t = Array(a)); ++n < a; )
    t[n] = e[n];
  return t;
}
function hr(e, t) {
  for (var n = -1, a = e == null ? 0 : e.length, s = 0, l = []; ++n < a; ) {
    var o = e[n];
    t(o, n, e) && (l[s++] = o);
  }
  return l;
}
function Gt() {
  return [];
}
var gr = Object.prototype, yr = gr.propertyIsEnumerable, st = Object.getOwnPropertySymbols, mr = st ? function(e) {
  return e == null ? [] : (e = Object(e), hr(st(e), function(t) {
    return yr.call(e, t);
  }));
} : Gt;
const Ke = mr;
function vr(e, t) {
  return xe(e, Ke(e), t);
}
function Ht(e, t) {
  for (var n = -1, a = t.length, s = e.length; ++n < a; )
    e[s + n] = t[n];
  return e;
}
var pr = Ft(Object.getPrototypeOf, Object);
const qt = pr;
var br = Object.getOwnPropertySymbols, $r = br ? function(e) {
  for (var t = []; e; )
    Ht(t, Ke(e)), e = qt(e);
  return t;
} : Gt;
const Wt = $r;
function wr(e, t) {
  return xe(e, Wt(e), t);
}
function Zt(e, t, n) {
  var a = t(e);
  return be(e) ? a : Ht(a, n(e));
}
function xr(e) {
  return Zt(e, De, Ke);
}
function kr(e) {
  return Zt(e, _e, Wt);
}
var Tr = Z(L, "DataView");
const Ie = Tr;
var Cr = Z(L, "Promise");
const Be = Cr;
var Vr = Z(L, "Set");
const Pe = Vr;
var jr = Z(L, "WeakMap");
const Oe = jr;
var rt = "[object Map]", Sr = "[object Object]", it = "[object Promise]", ut = "[object Set]", ct = "[object WeakMap]", dt = "[object DataView]", Ar = W(Ie), Ir = W(re), Br = W(Be), Pr = W(Pe), Or = W(Oe), G = Q;
(Ie && G(new Ie(new ArrayBuffer(1))) != dt || re && G(new re()) != rt || Be && G(Be.resolve()) != it || Pe && G(new Pe()) != ut || Oe && G(new Oe()) != ct) && (G = function(e) {
  var t = Q(e), n = t == Sr ? e.constructor : void 0, a = n ? W(n) : "";
  if (a)
    switch (a) {
      case Ar:
        return dt;
      case Ir:
        return rt;
      case Br:
        return it;
      case Pr:
        return ut;
      case Or:
        return ct;
    }
  return t;
});
const Fe = G;
var Mr = Object.prototype, Ur = Mr.hasOwnProperty;
function zr(e) {
  var t = e.length, n = new e.constructor(t);
  return t && typeof e[0] == "string" && Ur.call(e, "index") && (n.index = e.index, n.input = e.input), n;
}
var Er = L.Uint8Array;
const ft = Er;
function Ne(e) {
  var t = new e.constructor(e.byteLength);
  return new ft(t).set(new ft(e)), t;
}
function Lr(e, t) {
  var n = t ? Ne(e.buffer) : e.buffer;
  return new e.constructor(n, e.byteOffset, e.byteLength);
}
var Dr = /\w*$/;
function _r(e) {
  var t = new e.constructor(e.source, Dr.exec(e));
  return t.lastIndex = e.lastIndex, t;
}
var ht = R ? R.prototype : void 0, gt = ht ? ht.valueOf : void 0;
function Kr(e) {
  return gt ? Object(gt.call(e)) : {};
}
function Fr(e, t) {
  var n = t ? Ne(e.buffer) : e.buffer;
  return new e.constructor(n, e.byteOffset, e.length);
}
var Nr = "[object Boolean]", Rr = "[object Date]", Gr = "[object Map]", Hr = "[object Number]", qr = "[object RegExp]", Wr = "[object Set]", Zr = "[object String]", Yr = "[object Symbol]", Xr = "[object ArrayBuffer]", Jr = "[object DataView]", Qr = "[object Float32Array]", ei = "[object Float64Array]", ti = "[object Int8Array]", ni = "[object Int16Array]", ai = "[object Int32Array]", li = "[object Uint8Array]", oi = "[object Uint8ClampedArray]", si = "[object Uint16Array]", ri = "[object Uint32Array]";
function ii(e, t, n) {
  var a = e.constructor;
  switch (t) {
    case Xr:
      return Ne(e);
    case Nr:
    case Rr:
      return new a(+e);
    case Jr:
      return Lr(e, n);
    case Qr:
    case ei:
    case ti:
    case ni:
    case ai:
    case li:
    case oi:
    case si:
    case ri:
      return Fr(e, n);
    case Gr:
      return new a();
    case Hr:
    case Zr:
      return new a(e);
    case qr:
      return _r(e);
    case Wr:
      return new a();
    case Yr:
      return Kr(e);
  }
}
var yt = Object.create, ui = function() {
  function e() {
  }
  return function(t) {
    if (!de(t))
      return {};
    if (yt)
      return yt(t);
    e.prototype = t;
    var n = new e();
    return e.prototype = void 0, n;
  };
}();
const ci = ui;
function di(e) {
  return typeof e.constructor == "function" && !Le(e) ? ci(qt(e)) : {};
}
var fi = "[object Map]";
function hi(e) {
  return ee(e) && Fe(e) == fi;
}
var mt = J && J.isMap, gi = mt ? Ee(mt) : hi;
const yi = gi;
var mi = "[object Set]";
function vi(e) {
  return ee(e) && Fe(e) == mi;
}
var vt = J && J.isSet, pi = vt ? Ee(vt) : vi;
const bi = pi;
var $i = 1, wi = 2, xi = 4, Yt = "[object Arguments]", ki = "[object Array]", Ti = "[object Boolean]", Ci = "[object Date]", Vi = "[object Error]", Xt = "[object Function]", ji = "[object GeneratorFunction]", Si = "[object Map]", Ai = "[object Number]", Jt = "[object Object]", Ii = "[object RegExp]", Bi = "[object Set]", Pi = "[object String]", Oi = "[object Symbol]", Mi = "[object WeakMap]", Ui = "[object ArrayBuffer]", zi = "[object DataView]", Ei = "[object Float32Array]", Li = "[object Float64Array]", Di = "[object Int8Array]", _i = "[object Int16Array]", Ki = "[object Int32Array]", Fi = "[object Uint8Array]", Ni = "[object Uint8ClampedArray]", Ri = "[object Uint16Array]", Gi = "[object Uint32Array]", T = {};
T[Yt] = T[ki] = T[Ui] = T[zi] = T[Ti] = T[Ci] = T[Ei] = T[Li] = T[Di] = T[_i] = T[Ki] = T[Si] = T[Ai] = T[Jt] = T[Ii] = T[Bi] = T[Pi] = T[Oi] = T[Fi] = T[Ni] = T[Ri] = T[Gi] = !0;
T[Vi] = T[Xt] = T[Mi] = !1;
function ge(e, t, n, a, s, l) {
  var o, r = t & $i, i = t & wi, f = t & xi;
  if (n && (o = s ? n(e, a, s, l) : n(e)), o !== void 0)
    return o;
  if (!de(e))
    return e;
  var y = be(e);
  if (y) {
    if (o = zr(e), !r)
      return fr(e, o);
  } else {
    var x = Fe(e), m = x == Xt || x == ji;
    if (Lt(e))
      return dr(e, r);
    if (x == Jt || x == Yt || m && !s) {
      if (o = i || m ? {} : di(e), !r)
        return i ? wr(e, ur(o, e)) : vr(e, lr(o, e));
    } else {
      if (!T[x])
        return s ? e : {};
      o = ii(e, x, r);
    }
  }
  l || (l = new ne());
  var h = l.get(e);
  if (h)
    return h;
  l.set(e, o), bi(e) ? e.forEach(function(b) {
    o.add(ge(b, t, n, b, e, l));
  }) : yi(e) && e.forEach(function(b, j) {
    o.set(j, ge(b, t, n, j, e, l));
  });
  var g = f ? i ? kr : xr : i ? _e : De, v = y ? void 0 : g(e);
  return as(v || e, function(b, j) {
    v && (j = b, b = e[j]), Ut(o, j, ge(b, t, n, j, e, l));
  }), o;
}
var Hi = 1, qi = 4;
function me(e) {
  return ge(e, Hi | qi);
}
const Qt = (e) => ($t("data-v-6aca55dd"), e = e(), wt(), e), Wi = { class: "w-full flex flex-col" }, Zi = { class: "overflow-x-auto -mx-4 -my-2 lg:-mx-8 sm:-mx-6" }, Yi = { class: "inline-block min-w-full py-2 align-middle lg:px-8 md:px-6" }, Xi = {
  key: 0,
  class: "absolute left-12 top-0 h-12 flex items-center sm:left-16 space-x-3"
}, Ji = ["disabled", "onClick"], Qi = { class: "min-w-full table-fixed overflow-x-auto" }, eu = { class: "caption-top" }, tu = { class: "caption-bottom" }, nu = {
  key: 0,
  scope: "col",
  class: "relative w-12 px-6 sm:w-16 sm:px-8"
}, au = ["checked", "indeterminate"], lu = {
  key: 1,
  scope: "col"
}, ou = { class: "relative overflow-x-auto bg-white divide-y divide-gray-200" }, su = { colspan: 99 }, ru = { class: "absolute h-full w-full flex items-center justify-center" }, iu = /* @__PURE__ */ xt('<svg class="opacity-80" xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24" data-v-6aca55dd><g stroke="#888888" data-v-6aca55dd><circle cx="12" cy="12" r="9.5" fill="none" stroke-linecap="round" stroke-width="3" data-v-6aca55dd><animate attributeName="stroke-dasharray" calcMode="spline" dur="1.5s" keySplines="0.42,0,0.58,1;0.42,0,0.58,1;0.42,0,0.58,1" keyTimes="0;0.475;0.95;1" repeatCount="indefinite" values="0 150;42 150;42 150;42 150" data-v-6aca55dd></animate><animate attributeName="stroke-dashoffset" calcMode="spline" dur="1.5s" keySplines="0.42,0,0.58,1;0.42,0,0.58,1;0.42,0,0.58,1" keyTimes="0;0.475;0.95;1" repeatCount="indefinite" values="0;-16;-59;-59" data-v-6aca55dd></animate></circle><animateTransform attributeName="transform" dur="2s" repeatCount="indefinite" type="rotate" values="0 12 12;360 12 12" data-v-6aca55dd></animateTransform></g></svg>', 1), uu = { key: 0 }, cu = { colspan: 99 }, du = { class: "min-h-32 flex items-center justify-center" }, fu = /* @__PURE__ */ Qt(() => /* @__PURE__ */ c("svg", {
  class: "opacity-30",
  xmlns: "http://www.w3.org/2000/svg",
  width: "50",
  height: "50",
  viewBox: "0 0 24 24"
}, [
  /* @__PURE__ */ c("path", {
    fill: "#888888",
    d: "M12 11.997a1.316 1.316 0 1 1 0-2.632a1.316 1.316 0 0 1 0 2.632zm2.916-.021c0-2.828-1.109-5.397-2.916-7.298a10.556 10.556 0 0 0-2.916 7.298c0 1.297.234 2.535.66 3.683c-.618.9-1.074 2.16-1.275 3.616c.639-.767 1.422-1.306 2.292-1.591c.363.555.78 1.096 1.239 1.574c.461-.494.876-1.02 1.239-1.59c.87.271 1.653.826 2.29 1.576c-.199-1.456-.655-2.716-1.275-3.615c.427-1.155.66-2.385.66-3.69l.002.037zM12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10s-4.477 10-10 10zm0-22C5.373 0 0 5.373 0 12s5.373 12 12 12s12-5.373 12-12S18.627 0 12 0z"
  })
], -1)), hu = ["onMouseenter", "onClick"], gu = {
  key: 0,
  class: "relative w-12 px-6 sm:w-16 sm:px-8"
}, yu = {
  key: 0,
  class: "bg-primary-600 absolute inset-y-0 left-0 w-0.5"
}, mu = ["value", "indeterminate"], vu = ["onClick"], pu = ["onClick", "onMouseenter"], bu = {
  key: 0,
  class: "relative w-12 px-6 sm:w-16 sm:px-8"
}, $u = {
  key: 0,
  class: "bg-primary-600 absolute inset-y-0 left-0 w-0.5"
}, wu = ["value"], xu = /* @__PURE__ */ Qt(() => /* @__PURE__ */ c("div", { class: "i-heroicons:minus h-4 w-4 flex items-center opacity-20" }, null, -1)), ku = [
  xu
], Tu = {
  name: "UTable"
}, Cu = /* @__PURE__ */ C({
  ...Tu,
  props: {
    data: { default: () => [] },
    actions: { default: () => [] },
    loading: { type: Boolean, default: !1 },
    divided: { type: Boolean, default: !1 },
    bordered: { type: Boolean, default: !1 },
    striped: { type: Boolean, default: !1 },
    tree: { type: Boolean, default: !1 },
    headerColor: { type: Boolean, default: !1 },
    rowKey: { default: "id" },
    defaultExpandAll: { type: Boolean, default: !1 },
    rounded: { default: "sm" },
    associateParent: { default: "auto" }
  },
  emits: ["hoverRow", "clickRow"],
  setup(e) {
    const t = e, n = U(me(t.data)), a = U([]);
    q(() => t.data, (y) => {
      a.value = [], n.value = me(y), f();
    });
    const s = I(() => {
      let y = 0;
      return n.value.forEach((x) => {
        var m;
        y++, t.tree && ((m = x.children) != null && m.length) && x.children.forEach(() => y++);
      }), y;
    }), l = I(() => {
      const y = t.data;
      return y && y.length > 0 && a.value.length > 0 && a.value.length < s.value;
    });
    function o(y) {
      var m;
      const x = y.target;
      if (a.value = [], x.checked) {
        for (const h of n.value)
          if (a.value.push(h[t.rowKey]), t.tree && ((m = h.children) != null && m.length))
            for (const g of h.children)
              a.value.push(g[t.rowKey]);
      }
    }
    function r(y) {
      var x;
      if (t.tree && ((x = y.children) != null && x.length)) {
        let m = 0;
        for (const h of y.children)
          a.value.includes(h[t.rowKey]) && m++;
        return m > 0 && m < y.children.length;
      }
    }
    function i(y) {
      var g;
      if (!t.tree)
        return;
      const x = y.target;
      let m, h = !1;
      for (const v of n.value) {
        if (String(v[t.rowKey]) === x.value) {
          m = v, h = !0;
          break;
        }
        if ((g = v.children) != null && g.length) {
          for (const b of v.children)
            if (String(b[t.rowKey]) === x.value) {
              m = v, h = !1;
              break;
            }
        }
        if (m)
          break;
      }
      if (m != null && m.children.length)
        if (h)
          if (x.checked)
            for (const v of m.children)
              a.value.includes(v[t.rowKey]) || a.value.push(v[t.rowKey]);
          else
            for (const v of m.children)
              a.value = a.value.filter((b) => b !== v[t.rowKey]);
        else if (x.checked) {
          if (t.associateParent === "uncheck")
            return;
          if (t.associateParent === "check" && (a.value.includes(m[t.rowKey]) || a.value.push(m[t.rowKey])), t.associateParent === "auto") {
            let v = !0;
            for (const b of m.children)
              if (!a.value.includes(b[t.rowKey])) {
                v = !1;
                break;
              }
            v && !a.value.includes(m[t.rowKey]) && a.value.push(m[t.rowKey]);
          }
        } else {
          t.associateParent === "auto" && (a.value = a.value.filter((v) => v !== m[t.rowKey]));
          for (const v of m.children)
            if (a.value.includes(v[t.rowKey]))
              return;
          a.value = a.value.filter((v) => v !== m[t.rowKey]);
        }
    }
    function f() {
      t.tree && (n.value.forEach((y) => y._hideSubItem = !0), t.defaultExpandAll && n.value.forEach((y) => y._hideSubItem = !1));
    }
    return Me(() => {
      f();
    }), (y, x) => {
      var m, h;
      return u(), d("div", Wi, [
        c("div", Zi, [
          c("div", Yi, [
            c("div", {
              class: p(["relative overflow-hidden", [
                `rounded-${e.rounded}`,
                {
                  "ring-1 ring-gray-300 border-opacity-50": e.bordered
                }
              ]])
            }, [
              a.value.length > 0 ? (u(), d("div", Xi, [
                (u(!0), d(S, null, M(e.actions, (g) => (u(), d("button", {
                  key: g.name,
                  disabled: !a.value.length,
                  class: "focus:ring-primary-500 inline-flex items-center border border-gray-300 rounded bg-white px-2.5 py-1.5 text-xs font-medium text-gray-700 shadow-sm hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2",
                  onClick: (v) => g.clicked(a.value)
                }, A(g.name), 9, Ji))), 128))
              ])) : k("", !0),
              c("table", Qi, [
                c("caption", eu, [
                  $(y.$slots, "top", {}, void 0, !0)
                ]),
                c("caption", tu, [
                  $(y.$slots, "bottom", {}, void 0, !0)
                ]),
                c("thead", {
                  class: p({ "bg-gray-50": e.headerColor })
                }, [
                  c("tr", {
                    class: p({ "divide-x divide-gray-300": e.divided })
                  }, [
                    (m = e.actions) != null && m.length ? (u(), d("th", nu, [
                      c("input", {
                        type: "checkbox",
                        class: "checkbox",
                        checked: ((h = n.value) == null ? void 0 : h.length) > 0 && a.value.length === w(s),
                        indeterminate: w(l),
                        onChange: o
                      }, null, 40, au)
                    ])) : k("", !0),
                    e.tree ? (u(), d("th", lu)) : k("", !0),
                    $(y.$slots, "headers", {}, void 0, !0)
                  ], 2)
                ], 2),
                c("tbody", ou, [
                  c("tr", {
                    class: p([{ invisible: !e.loading }, "absolute h-full w-full"])
                  }, [
                    c("td", su, [
                      c("div", ru, [
                        $(y.$slots, "loading", {}, () => [
                          iu
                        ], !0)
                      ])
                    ])
                  ], 2),
                  n.value.length === 0 ? (u(), d("tr", uu, [
                    c("td", cu, [
                      c("div", du, [
                        $(y.$slots, "empty", {}, () => [
                          fu
                        ], !0)
                      ])
                    ])
                  ])) : (u(!0), d(S, { key: 1 }, M(n.value, (g, v) => {
                    var b;
                    return u(), d(S, {
                      key: g[e.rowKey]
                    }, [
                      c("tr", {
                        class: p(["duration-300 hover:bg-gray-100", {
                          "even:bg-gray-50": e.striped,
                          "divide-x divide-gray-300": e.divided,
                          "bg-gray-50": a.value.includes(g[e.rowKey]),
                          "opacity-30": e.loading
                          // 'bg-gray-100': item.children?.length,
                        }]),
                        onMouseenter: (j) => y.$emit("hoverRow", g),
                        onClick: D((j) => y.$emit("clickRow", g), ["stop"])
                      }, [
                        (b = e.actions) != null && b.length ? (u(), d("td", gu, [
                          a.value.includes(g[e.rowKey]) ? (u(), d("div", yu)) : k("", !0),
                          O(c("input", {
                            "onUpdate:modelValue": x[0] || (x[0] = (j) => a.value = j),
                            value: g[e.rowKey],
                            type: "checkbox",
                            class: "checkbox",
                            indeterminate: e.associateParent !== "uncheck" ? r(g) : !1,
                            onChange: i,
                            onClick: x[1] || (x[1] = D(() => {
                            }, ["stop"]))
                          }, null, 40, mu), [
                            [se, a.value]
                          ])
                        ])) : k("", !0),
                        e.tree ? (u(), d("td", {
                          key: 1,
                          class: p(["w-1", { "pl-4": !e.actions.length }])
                        }, [
                          c("button", {
                            class: p(["i-heroicons:chevron-right h-4 w-4 flex items-center duration-200", { "rotate-90": !g._hideSubItem }]),
                            onClick: D((j) => g._hideSubItem = !g._hideSubItem, ["stop"])
                          }, null, 10, vu)
                        ], 2)) : k("", !0),
                        $(y.$slots, "rows", {
                          row: g,
                          index: v
                        }, void 0, !0)
                      ], 42, hu),
                      e.tree && !g._hideSubItem ? (u(!0), d(S, { key: 0 }, M(g.children, (j, z) => {
                        var Re;
                        return u(), d("tr", {
                          key: j[e.rowKey],
                          class: p(["hover:bg-gray-100", {
                            "bg-gray-50": a.value.includes(g[e.rowKey]),
                            "opacity-30": e.loading
                          }]),
                          onClick: D((ke) => y.$emit("clickRow", j), ["stop"]),
                          onMouseenter: (ke) => y.$emit("hoverRow", j)
                        }, [
                          (Re = e.actions) != null && Re.length ? (u(), d("td", bu, [
                            a.value.includes(j[e.rowKey]) ? (u(), d("div", $u)) : k("", !0),
                            O(c("input", {
                              "onUpdate:modelValue": x[2] || (x[2] = (ke) => a.value = ke),
                              value: j[e.rowKey],
                              type: "checkbox",
                              class: "checkbox",
                              onChange: i,
                              onClick: x[3] || (x[3] = D(() => {
                              }, ["stop"]))
                            }, null, 40, wu), [
                              [se, a.value]
                            ])
                          ])) : k("", !0),
                          c("td", {
                            class: p(["w-1", { "pl-4": !e.actions.length }])
                          }, ku, 2),
                          $(y.$slots, "subs", {
                            sub: j,
                            index: z
                          }, void 0, !0)
                        ], 42, pu);
                      }), 128)) : k("", !0)
                    ], 64);
                  }), 128))
                ])
              ])
            ], 2)
          ])
        ])
      ]);
    };
  }
});
const Vu = /* @__PURE__ */ ce(Cu, [["__scopeId", "data-v-6aca55dd"]]), ju = {
  name: "UToast"
}, Su = /* @__PURE__ */ C({
  ...ju,
  props: {
    show: { type: Boolean },
    position: { default: "top" },
    align: { default: "center" },
    timeout: { default: 2500 },
    queue: { type: Boolean, default: !0 },
    zIndex: { default: 100 },
    closeable: { type: Boolean, default: !1 }
  },
  setup(e, { expose: t }) {
    const n = e, a = F({
      events: [],
      success: (s) => a.add("success", s),
      info: (s) => a.add("info", s),
      warning: (s) => a.add("warning", s),
      error: (s) => a.add("error", s),
      show: (s, l) => a.add(s, l),
      add: (s, l) => {
        n.queue || (a.events = []), setTimeout(() => {
          const o = { id: Date.now(), content: l, type: s };
          a.events.push(o), setTimeout(() => a.remove(o), n.timeout);
        }, 100);
      },
      remove: (s) => {
        a.events = a.events.filter((l) => l.id !== s.id);
      }
    });
    return t({
      show: a.show,
      success: a.success,
      info: a.info,
      warning: a.warning,
      error: a.error
    }), (s, l) => O((u(), d("div", {
      class: p(["fixed w-4/5 sm:w-96", {
        "left-1/2 -translate-x-1/2": e.align === "center",
        "left-6": e.align === "left",
        "right-6": e.align === "right",
        "top-4": e.position === "top",
        "bottom-6 ": e.position === "bottom"
      }]),
      style: ve({ zIndex: e.zIndex })
    }, [
      K(hn, {
        tag: "ul",
        "enter-active-class": "transition ease-out duration-200",
        "leave-active-class": "transition ease-in duration-200 absolute w-full",
        "enter-class": e.position === "bottom" ? "transform translate-y-3 opacity-0" : "transform -translate-y-3 opacity-0",
        "enter-to-class": "transform translate-y-0 opacity-100",
        "leave-class": "transform translate-y-0 opacity-100",
        "leave-to-class": e.position === "bottom" ? "transform translate-y-1/4 opacity-0" : "transform -translate-y-1/4 opacity-0",
        "move-class": "ease-in-out duration-200",
        class: "inline-block w-full"
      }, {
        default: B(() => [
          (u(!0), d(S, null, M(a.events, (o) => (u(), d("li", {
            key: o.id,
            class: p({
              "pb-2": e.position === "bottom",
              "pt-2": e.position === "top"
            })
          }, [
            $(s.$slots, "default", {
              type: o.type,
              content: o.content
            }, () => [
              K(Tt, {
                type: o.type,
                closeable: e.closeable,
                onClose: (r) => a.remove(o)
              }, {
                default: B(() => [
                  _(A(o.content), 1)
                ]),
                _: 2
              }, 1032, ["type", "closeable", "onClose"])
            ])
          ], 2))), 128))
        ]),
        _: 3
      }, 8, ["enter-class", "leave-to-class"])
    ], 6)), [
      [kt, a.events.length]
    ]);
  }
}), Au = { class: "w-full" }, Iu = { class: "hover:bg-primary-100 flex cursor-pointer items-center gap-1 px-2 py-0.5 transition-100 hover:rounded" }, Bu = ["onClick"], Pu = /* @__PURE__ */ c("svg", {
  xmlns: "http://www.w3.org/2000/svg",
  fill: "none",
  viewBox: "0 0 24 24",
  "stroke-width": "1.5",
  stroke: "currentColor"
}, [
  /* @__PURE__ */ c("path", {
    "stroke-linecap": "round",
    "stroke-linejoin": "round",
    d: "M19.5 8.25l-7.5 7.5-7.5-7.5"
  })
], -1), Ou = [
  Pu
], Mu = {
  key: 0,
  class: "flex items-center"
}, Uu = ["disabled", "value", "indeterminate"], zu = { key: 1 }, Eu = { class: "pl-4" }, Lu = {
  name: "UTree"
}, Du = /* @__PURE__ */ C({
  ...Lu,
  props: {
    modelValue: { default: null },
    options: { default: () => [] },
    cascade: { type: Boolean, default: !1 },
    associateParent: { default: "auto" },
    selectable: { type: Boolean, default: !1 },
    defaultExpandedKeys: { default: () => [] },
    level: { default: 1 }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const n = e, a = U(me(n.options)), s = gn();
    q(() => n.options, (h) => {
      a.value = me(h);
    });
    const l = I({
      get: () => n.modelValue,
      set: (h) => t("update:modelValue", h)
    });
    function o(h) {
      var g;
      if (n.cascade && ((g = h.children) != null && g.length)) {
        let v = 0;
        for (const b of h.children)
          l.value.includes(b.value) && v++;
        return v > 0 && v < h.children.length;
      }
    }
    function r(h, g) {
      if (g) {
        for (const v of g)
          if (v.children) {
            if (v.children.some((b) => String(b.value) === h))
              return v;
            {
              const b = r(h, v.children);
              if (b)
                return b;
            }
          }
      }
    }
    function i(h) {
      var g;
      if (n.associateParent !== "uncheck") {
        if (n.associateParent === "auto") {
          const v = r(h, s["all-options"]);
          v && ((g = v.children) != null && g.every((j) => l.value.includes(j.value)) && !l.value.includes(v.value) && l.value.push(v.value), i(v.value));
        }
        if (n.associateParent === "check") {
          const v = r(h, s["all-options"]);
          v && (l.value.includes(v.value) || l.value.push(v.value), i(v.value));
        }
      }
    }
    function f(h) {
      const g = r(h, s["all-options"]);
      g && (l.value.includes(g.value) && l.value.splice(l.value.indexOf(g.value), 1), f(g.value));
    }
    function y(h) {
      if (!n.cascade)
        return;
      const { value: g, checked: v } = h.target;
      v ? i(g) : f(g), x(
        a.value.find((b) => String(b.value) === g),
        l.value.includes(g) || l.value.includes(Number(g))
      );
    }
    function x(h, g) {
      !h || h.disabled || (g ? l.value.includes(h.value) || l.value.push(h.value) : l.value.includes(h.value) && l.value.splice(l.value.indexOf(h.value), 1), n.cascade && h.children && h.children.forEach((v) => x(v, g)));
    }
    function m(h) {
      var g;
      h.hidden = !0, n.defaultExpandedKeys.length && (n.defaultExpandedKeys.includes(h.value) && (h.hidden = !1), (g = h.children) == null || g.forEach((v) => m(v)));
    }
    return Me(() => {
      a.value.forEach((h) => m(h));
    }), (h, g) => {
      const v = yn("UTree");
      return u(), d("div", Au, [
        (u(!0), d(S, null, M(a.value, (b, j) => (u(), d("div", {
          key: b.value + j
        }, [
          c("div", Iu, [
            c("span", {
              class: p(["mr-1 h-4 w-4 cursor-pointer text-gray-500 duration-200", {
                invisible: !b.children,
                "-rotate-90": b.hidden
              }]),
              onClick: D((z) => b.hidden = !b.hidden, ["stop"])
            }, Ou, 10, Bu),
            e.selectable ? (u(), d("div", Mu, [
              O(c("input", {
                "onUpdate:modelValue": g[0] || (g[0] = (z) => Y(l) ? l.value = z : null),
                type: "checkbox",
                disabled: b.disabled,
                value: b.value,
                class: p(["focus:ring-primary-500 text-primary-400 border-gray-300 rounded", b.disabled ? "cursor-not-allowed opacity-30" : "cursor-pointer"]),
                indeterminate: e.associateParent !== "uncheck" ? o(b) : !1,
                onChange: y
              }, null, 42, Uu), [
                [se, w(l)]
              ]),
              c("label", {
                class: p(["ml-2", b.disabled ? "cursor-not-allowed" : "cursor-pointer"])
              }, [
                $(h.$slots, "option", {
                  item: b,
                  level: e.level
                }, () => [
                  _(A(b.label), 1)
                ])
              ], 2)
            ])) : (u(), d("span", zu, [
              $(h.$slots, "option", { item: b }, () => [
                _(A(b.label), 1)
              ])
            ]))
          ]),
          K(pe, null, {
            default: B(() => [
              O(c("div", Eu, [
                K(v, {
                  modelValue: w(l),
                  "onUpdate:modelValue": g[1] || (g[1] = (z) => Y(l) ? l.value = z : null),
                  cascade: e.cascade,
                  selectable: e.selectable,
                  options: b.children || [],
                  "default-expanded-keys": e.defaultExpandedKeys,
                  level: e.level + 1,
                  "all-options": e.level === 1 ? e.options : h.$attrs["all-options"],
                  "associate-parent": e.associateParent
                }, {
                  option: B((z) => [
                    $(h.$slots, "option", {
                      item: z.item,
                      level: z.level
                    }, () => [
                      _(A(z.item.label), 1)
                    ])
                  ]),
                  _: 2
                }, 1032, ["modelValue", "cascade", "selectable", "options", "default-expanded-keys", "level", "all-options", "associate-parent"])
              ], 512), [
                [kt, !b.hidden]
              ])
            ]),
            _: 2
          }, 1024)
        ]))), 128))
      ]);
    };
  }
}), _u = { class: "w-full px-3" }, Ku = { class: "flex items-center" }, Fu = {
  key: 0,
  class: "mr-3 text-gray-600"
}, Nu = ["onClick"], Ru = { class: "p-3" }, Gu = {
  name: "UTabs"
}, Hu = /* @__PURE__ */ C({
  ...Gu,
  props: {
    modelValue: { default: null },
    full: { type: Boolean, default: !1 },
    type: { default: "card" }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    var r;
    const n = e;
    mn("tabs", { value: n.modelValue });
    const a = vn(), s = (r = a.default) == null ? void 0 : r.call(a), l = F({
      tab: [],
      activeValue(i, f) {
        if (typeof n.modelValue == "number")
          return f;
        if (typeof n.modelValue == "string")
          return i.value;
      },
      selectTab(i, f) {
        typeof n.modelValue == "number" && t("update:modelValue", f), typeof n.modelValue == "string" && t("update:modelValue", i.value);
      }
    });
    if (s)
      for (let i = 0; i < s.length; i++)
        l.tab.push(s[i].props);
    function o() {
      var i;
      if (a.default) {
        if (typeof n.modelValue == "number")
          return a.default()[n.modelValue];
        if (typeof n.modelValue == "string" && s) {
          for (let f = 0; f < s.length; f++)
            if (((i = s[f].props) == null ? void 0 : i.value) === n.modelValue)
              return a.default()[f];
        }
      }
    }
    return (i, f) => (u(), d("div", _u, [
      c("div", Ku, [
        i.$slots.prefix ? (u(), d("div", Fu, [
          $(i.$slots, "prefix", {}, void 0, !0)
        ])) : k("", !0),
        (u(!0), d(S, null, M(l.tab, (y, x) => (u(), d("div", {
          key: x,
          class: p(["tab", {
            active: l.activeValue(y, x) === e.modelValue,
            "w-full justify-center": e.type === "flex"
          }]),
          onClick: (m) => l.selectTab(y, x)
        }, [
          c("span", null, A(y == null ? void 0 : y.title), 1),
          c("span", {
            class: p(["ml-1 inline-block h-5 w-5", y == null ? void 0 : y.icon])
          }, null, 2)
        ], 10, Nu))), 128))
      ]),
      c("div", Ru, [
        K(o)
      ])
    ]));
  }
});
const qu = /* @__PURE__ */ ce(Hu, [["__scopeId", "data-v-1641114e"]]), Wu = {
  name: "UTabPane"
}, Zu = /* @__PURE__ */ C({
  ...Wu,
  props: {
    title: { default: "" },
    icon: { default: void 0 },
    value: { default: void 0 }
  },
  setup(e) {
    return (t, n) => (u(), d("div", Ve(pn(t.$attrs)), [
      $(t.$slots, "default")
    ], 16));
  }
}), en = [
  Tt,
  Vn,
  fe,
  Qn,
  oa,
  ua,
  ha,
  Ca,
  Da,
  Za,
  sl,
  hl,
  wl,
  Tl,
  Il,
  Ml,
  ql,
  Zl,
  Ae,
  so,
  Vu,
  Se,
  Su,
  Du,
  qu,
  Zu
];
for (const e of en)
  e.install = (t) => {
    t.component(e.name, e);
  };
function Yu(e) {
  en.forEach((t) => {
    e.component(t.name, t);
  });
}
const Ju = {
  install: Yu
};
export {
  Tt as UAlert,
  Vn as UBadge,
  fe as UButton,
  Qn as UCheckbox,
  oa as UCollapse,
  ua as UDivider,
  ha as UDrawer,
  Ca as UDropdown,
  Da as UDynamicTags,
  Za as UFileInput,
  sl as UInput,
  hl as ULoading,
  wl as UModal,
  Tl as UPopover,
  Il as UProgress,
  Ml as URadio,
  ql as USelect,
  Zl as USkeleton,
  Ae as USpin,
  so as USwitch,
  Zu as UTabPane,
  Vu as UTable,
  qu as UTabs,
  Se as UTag,
  Su as UToast,
  Du as UTree,
  Ju as default
};
