// vite-plugin-vue-devtools@0.0.16/dist/client/assets/__inspecting-dba9bf6f.js
import { _ as __unplugin_components_0 } from './VPanelGrids-e1d9f019.js';
import { v as defineComponent, aC as useEventListener, o as openBlock, c as createBlock, h as withCtx, x as createBaseVNode, aD as useDevtoolsClient } from './index-d1c9f78d.js';

const _hoisted_1 = /* @__PURE__ */ createBaseVNode("div", null, " Inspecting Vue components ", -1);
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "__inspecting",
  setup(__props) {
    const client = useDevtoolsClient();
    useEventListener("keydown", (e) => {
      if (e.key === "Escape")
        close();
    });
    function close() {
      client.value?.inspector?.disable();
    }
    return (_ctx, _cache) => {
      const _component_VPanelGrids = __unplugin_components_0;
      return openBlock(), createBlock(_component_VPanelGrids, {
        "h-screen": "",
        "w-screen": "",
        px5: ""
      }, {
        default: withCtx(() => [
          createBaseVNode("div", {
            absolute: "",
            "right-0": "",
            "top-0": "",
            p2: ""
          }, [
            createBaseVNode("button", {
              "carbon-close": "",
              ma: "",
              "text-xl": "",
              op50: "",
              "hover:op100": "",
              onClick: close
            })
          ]),
          _hoisted_1
        ]),
        _: 1
      });
    };
  }
});

/* Injected with object hook! */

/* Injected with object hook! */

export { _sfc_main as default };
