// vite-plugin-vue-devtools@0.0.16/dist/client/assets/routes-671211d0.js
import { _ as __unplugin_components_0 } from './VPanelGrids-e1d9f019.js';
import { _ as __unplugin_components_7 } from './VCard-b4cd9ab3.js';
import { v as defineComponent, z as unref, T as router, o as openBlock, e as createElementBlock, u as createVNode, h as withCtx, x as createBaseVNode, F as Fragment, i as renderList, U as routeRecordMatcherState, V as activeRouteRecordMatcherState, c as createBlock, S as _sfc_main$2, q as normalizeClass, W as activeRouteRecordIndex, X as toggleRouteRecordMatcher, O as createTextVNode, y as toDisplayString, l as normalizeStyle, g as createCommentVNode, Y as __unplugin_components_1 } from './index-d1c9f78d.js';
import { g, M, _ as _sfc_main$1 } from './splitpanes.es-330c3706.js';
import './VIcon.vue_vue_type_script_setup_true_lang-e30540ed.js';
import './index-45ea79cd.js';
import './VIconButton.vue_vue_type_script_setup_true_lang-c612d7f4.js';

const _hoisted_1 = {
  key: 0,
  "h-screen": "",
  "n-panel-grids": ""
};
const _hoisted_2 = {
  "h-screen": "",
  "select-none": "",
  "overflow-scroll": "",
  "p-2": "",
  class: "no-scrollbar"
};
const _hoisted_3 = ["onClick"];
const _hoisted_4 = { "vue-block-title": "" };
const _hoisted_5 = {
  "h-screen": "",
  "select-none": "",
  "overflow-scroll": "",
  "p-2": "",
  class: "no-scrollbar"
};
const _hoisted_6 = /* @__PURE__ */ createBaseVNode("h1", { "text-xl": "" }, " Install Vue Router ", -1);
const _hoisted_7 = /* @__PURE__ */ createBaseVNode("p", {
  "text-sm": "",
  op50: ""
}, " It seems you don't have vue-router installed. ", -1);
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "routes",
  setup(__props) {
    return (_ctx, _cache) => {
      const _component_VBadge = __unplugin_components_1;
      const _component_StateFields = _sfc_main$1;
      const _component_TabIcon = _sfc_main$2;
      const _component_VCard = __unplugin_components_7;
      const _component_VPanelGrids = __unplugin_components_0;
      return unref(router) ? (openBlock(), createElementBlock("div", _hoisted_1, [
        createVNode(unref(M), null, {
          default: withCtx(() => [
            createVNode(unref(g), { border: "r base" }, {
              default: withCtx(() => [
                createBaseVNode("div", _hoisted_2, [
                  (openBlock(true), createElementBlock(Fragment, null, renderList(unref(routeRecordMatcherState), (item, index) => {
                    return openBlock(), createElementBlock("div", {
                      key: index,
                      "vue-block": "",
                      class: normalizeClass([unref(activeRouteRecordIndex) === index ? "vue-block-active" : "vue-block-hover"]),
                      onClick: ($event) => unref(toggleRouteRecordMatcher)(index)
                    }, [
                      createBaseVNode("h3", _hoisted_4, [
                        createBaseVNode("span", {
                          class: normalizeClass([unref(activeRouteRecordIndex) === index ? "text-white" : "vue-block-text"])
                        }, [
                          createTextVNode(toDisplayString(item.path) + " ", 1),
                          (openBlock(true), createElementBlock(Fragment, null, renderList(item.tags, (tag, childIndex) => {
                            return openBlock(), createElementBlock(Fragment, null, [
                              tag.label ? (openBlock(), createBlock(_component_VBadge, {
                                key: childIndex,
                                "text-black": "",
                                style: normalizeStyle({
                                  backgroundColor: `#${tag.bgColor.toString(16)}`
                                })
                              }, {
                                default: withCtx(() => [
                                  createTextVNode(toDisplayString(tag.label), 1)
                                ]),
                                _: 2
                              }, 1032, ["style"])) : createCommentVNode("", true)
                            ], 64);
                          }), 256))
                        ], 2)
                      ])
                    ], 10, _hoisted_3);
                  }), 128))
                ])
              ]),
              _: 1
            }),
            createVNode(unref(g), null, {
              default: withCtx(() => [
                createBaseVNode("div", _hoisted_5, [
                  createVNode(_component_StateFields, { data: unref(activeRouteRecordMatcherState) }, null, 8, ["data"])
                ])
              ]),
              _: 1
            })
          ]),
          _: 1
        })
      ])) : (openBlock(), createBlock(_component_VPanelGrids, {
        key: 1,
        px5: ""
      }, {
        default: withCtx(() => [
          createVNode(_component_VCard, {
            flex: "~ col gap2",
            "min-w-30": "",
            "items-center": "",
            p3: ""
          }, {
            default: withCtx(() => [
              createVNode(_component_TabIcon, {
                mb2: "",
                "text-5xl": "",
                icon: "i-logos-vue",
                title: "Vue Router"
              }),
              _hoisted_6,
              _hoisted_7
            ]),
            _: 1
          })
        ]),
        _: 1
      }));
    };
  }
});

/* Injected with object hook! */

/* Injected with object hook! */

export { _sfc_main as default };
