// vite-plugin-vue-devtools@0.0.16/dist/client/assets/VTextInput.vue_vue_type_script_setup_true_lang-707de251.js
import { _ as _sfc_main$2 } from './VIconTitle.vue_vue_type_script_setup_true_lang-5762ce49.js';
import { _ as _sfc_main$3 } from './VIcon.vue_vue_type_script_setup_true_lang-e30540ed.js';
import { v as defineComponent, I as useVModel, o as openBlock, e as createElementBlock, x as createBaseVNode, q as normalizeClass, u as createVNode, h as withCtx, f as renderSlot, g as createCommentVNode, c as createBlock, z as unref, d as withDirectives, F as Fragment, b as resolveDirective, O as createTextVNode, y as toDisplayString, p as pushScopeId, a as popScopeId, ax as _export_sfc, ay as vModelDynamic, k as mergeProps, K as isRef } from './index-d1c9f78d.js';

const _withScopeId = (n) => (pushScopeId("data-v-cf2e46eb"), n = n(), popScopeId(), n);
const _hoisted_1$1 = ["open"];
const _hoisted_2 = { "text-base": "" };
const _hoisted_3 = {
  key: 0,
  "text-sm": "",
  op50: ""
};
const _hoisted_4 = /* @__PURE__ */ _withScopeId(() => /* @__PURE__ */ createBaseVNode("div", { class: "flex-auto" }, null, -1));
const _hoisted_5 = /* @__PURE__ */ _withScopeId(() => /* @__PURE__ */ createBaseVNode("div", { class: "x-divider" }, null, -1));
const _sfc_main$1 = /* @__PURE__ */ defineComponent({
  __name: "VSectionBlock",
  props: {
    icon: null,
    text: null,
    description: null,
    containerClass: { default: "" },
    collapse: { type: Boolean, default: true },
    open: { type: Boolean, default: true },
    padding: { type: [Boolean, String], default: true }
  },
  setup(__props) {
    const props = __props;
    const open = useVModel(props, "open", void 0, { passive: true });
    function onToggle(e) {
      open.value = e.target.open;
    }
    return (_ctx, _cache) => {
      const _component_VIcon = _sfc_main$3;
      const _component_VIconTitle = _sfc_main$2;
      const _directive_lazy_show = resolveDirective("lazy-show");
      return openBlock(), createElementBlock(Fragment, null, [
        createBaseVNode("details", {
          open: unref(open),
          onToggle
        }, [
          createBaseVNode("summary", {
            class: normalizeClass(["cursor-pointer select-none hover:bg-active p4", __props.collapse ? "" : "pointer-events-none"])
          }, [
            createVNode(_component_VIconTitle, {
              icon: __props.icon,
              text: __props.text,
              "text-xl": "",
              transition: "",
              class: normalizeClass(unref(open) ? "op100" : "op60")
            }, {
              default: withCtx(() => [
                createBaseVNode("div", null, [
                  createBaseVNode("div", _hoisted_2, [
                    renderSlot(_ctx.$slots, "text", {}, () => [
                      createTextVNode(toDisplayString(__props.text), 1)
                    ], true)
                  ]),
                  __props.description || _ctx.$slots.description ? (openBlock(), createElementBlock("div", _hoisted_3, [
                    renderSlot(_ctx.$slots, "description", {}, () => [
                      createTextVNode(toDisplayString(__props.description), 1)
                    ], true)
                  ])) : createCommentVNode("", true)
                ]),
                _hoisted_4,
                renderSlot(_ctx.$slots, "actions", {}, void 0, true),
                __props.collapse ? (openBlock(), createBlock(_component_VIcon, {
                  key: 0,
                  icon: "carbon-chevron-down",
                  class: "chevron",
                  "cursor-pointer": "",
                  "place-self-start": "",
                  "text-base": "",
                  op75: "",
                  transition: "",
                  "duration-500": ""
                })) : createCommentVNode("", true)
              ]),
              _: 3
            }, 8, ["icon", "text", "class"])
          ], 2),
          withDirectives((openBlock(), createElementBlock("div", {
            class: normalizeClass(["flex flex-col flex-gap2 pb6 pt2", typeof __props.padding === "string" ? __props.padding : __props.padding ? "px4" : ""])
          }, [
            renderSlot(_ctx.$slots, "details", {}, void 0, true),
            createBaseVNode("div", {
              class: normalizeClass([__props.containerClass, "mt1"])
            }, [
              renderSlot(_ctx.$slots, "default", {}, void 0, true)
            ], 2),
            renderSlot(_ctx.$slots, "footer", {}, void 0, true)
          ], 2)), [
            [_directive_lazy_show, unref(open)]
          ])
        ], 40, _hoisted_1$1),
        _hoisted_5
      ], 64);
    };
  }
});

/* Injected with object hook! */

/* unplugin-vue-components disabled */const VSectionBlock_vue_vue_type_style_index_0_scoped_cf2e46eb_lang = '';
/* Injected with object hook! */

const __unplugin_components_3 = /* @__PURE__ */ _export_sfc(_sfc_main$1, [["__scopeId", "data-v-cf2e46eb"]]);

/* Injected with object hook! */

const _hoisted_1 = { class: "n-text-input flex flex items-center border n-border-base rounded py-1 pl-1 pr-2 focus-within:n-focus-base focus-within:border-context n-bg-base" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "VTextInput",
  props: {
    modelValue: { default: "" },
    icon: null,
    placeholder: null,
    disabled: { type: Boolean },
    type: { default: "text" }
  },
  setup(__props, { emit }) {
    const props = __props;
    const input = useVModel(props, "modelValue", emit, { passive: true });
    return (_ctx, _cache) => {
      const _component_VIcon = _sfc_main$3;
      return openBlock(), createElementBlock("div", _hoisted_1, [
        renderSlot(_ctx.$slots, "icon", {}, () => [
          __props.icon ? (openBlock(), createBlock(_component_VIcon, {
            key: 0,
            icon: __props.icon,
            class: "ml-0.3em mr-0.1em text-1.1em op50"
          }, null, 8, ["icon"])) : createCommentVNode("", true)
        ]),
        withDirectives(createBaseVNode("input", mergeProps({
          "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => isRef(input) ? input.value = $event : null)
        }, _ctx.$props, { class: "ml-0.4em w-full flex-auto n-bg-base !outline-none" }), null, 16), [
          [vModelDynamic, unref(input)]
        ])
      ]);
    };
  }
});

/* Injected with object hook! */

export { _sfc_main as _, __unplugin_components_3 as a };
